---
title: "Workshop Outline and Objectives"
author:
  - name: Jermiah J. Joseph
    email: jermiah.joseph@uhn.ca
    affiliation:
        - &pm Princess Margaret Cancer Centre
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: false
    toc_depth: 2
    code_folding: show
date: "`r doc_date()`"
package: "`r pkg_ver('CBWWorkshop2024')`"
vignette: >
  %\VignetteIndexEntry{Workshop Outline and Objectives}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    crop = NULL ## Related to https://stat.ethz.ch/pipermail/bioc-devel/2020-April/016656.html
)
```


```{r vignetteSetup, echo=FALSE, message=FALSE, warning = FALSE}
## Track time spent on making the vignette
startTime <- Sys.time()

## Bib setup
library("RefManageR")

## Write bibliography information
bib <- c(
    R = citation(),
    BiocStyle = citation("BiocStyle")[1],
    knitr = citation("knitr")[1],
    RefManageR = citation("RefManageR")[1],
    rmarkdown = citation("rmarkdown")[1],
    sessioninfo = citation("sessioninfo")[1],
    testthat = citation("testthat")[1],
    CBWWorkshop2024 = citation("CBWWorkshop2024")[1]
)
```

# Course Description {-}

This course explores the evolving field of personalized cancer treatment, focusing on tailoring treatment strategies to patients based on their unique molecular profiles. Participants will gain a comprehensive understanding of the field of pharmacogenomics and a guided workflow covering basic to advanced biomarker analysis. Our primary emphasis is on the analysis of cancer biomarkers; however, the principles and methodologies covered in this context are transferable and applicable to other disease areas.

The course is designed to provide sufficient context to beginners, building up from basics to tools and methods of data analysis. The hands-on practical tutorials for each session will provide the participants with practical insights into the data as well as tools to perform biomarker analysis on their own. By the end of this introductory course, participants will be equipped with the knowledge and skills to understand data requirements and perform biomarker analysis. 

# Course Objectives
Participants will gain practical experience and skills to be able to:

Extract features for biomarker discovery from -omics data (RNA-Seq, methylation, mutations, proteomics, chromatin profiling, etc)
Navigate pharmacogenomics datasets and data preparation for pharmacogenomic analysis
Identify univariable biomarkers from RNA-Seq data
Perform meta-analysis to identify robust biomarkers
Visualize and summarize the output of pharmacogenomic biomarker analysis

# Prerequisites
Participants should be comfortable writing and reading basic R.

You will require your own laptop computer. Minimum requirements: 1024×768 screen resolution, 1.5GHz CPU, 4GB RAM, 20GB free disk space, recent versions of Windows, Mac OS X or Linux (Most computers purchased in the past 3-4 years likely meet these requirements).

This workshop requires participants to complete pre-workshop tasks and readings which includes installing and setting up an R environment. 


# Course Outline


## Module 1: Introduction to Pharmacogenomics and Biomarker Discovery

### Lecture: 

* A brief introduction to precision oncology and biomarkers
  * Overview to personalized cancer treatment and the role of biomarker discovery
  * What is “pharmacogenomics”; is it different from pharmacogenetics?
  * Challenges in biomarker discovery 

* Brief introduction to biomarkers in pharmacogenomics
  * Conventional markers: oncogenic mutations, gene expression, proteomics
  * Exciting new fields: mutational signatures, synthetic lethality, chromatin accessibility
  * Other markers like radiomic features
  
* Overview of methods for patient profiling
  * Sequencing technologies: RNA-Seq, Whole Genome Sequencing (WGS), Whole Exome Sequencing (WES), etc.
  * Solid tumor vs liquid biopsy
  
### Practical – Getting to know multi-omics data 
TAs: Julia Nguyen, Jermiah Joseph

* Loading in expression matrices for various -omics data (RNA-Seq, methylation, mutations, proteomics, chromatin profiling) 
* Exploring feature extraction techniques to define biomarkers from (epi)genomic data
  
## Module 2: Data Preparation for Pharmacogenomic Analysis

### Lecture: 

* Common preclinical pharmacogenomic models and importance of clinical validation
  * Cell lines, patient-derived xenografts (PDX), organoids, etc.
  * Importance of clinical validation in biomarker discovery
* What is in a typical pharmacogenomic dataset?: dataset overview using PharmacoGx
* Quality control: batch effects, outliers, missing values, and common sources of variation

### Practical: Hands-on with pharmacogenomics data

* Loading a PharmacoSet in R using PharmacoGx or ORCESTRA, understanding the data structure, accessing features, and preparing metadata
* Plot batch effects using PCA and run correction method (SVA)
* Filter out outliers and missing values

## Module 3: Pharmacogenomics for biomarker discovery – Basic analysis

### Lecture:

* Data analysis walk-through
  * Statistical methods for drug response associations
  * Computing drug response associations using Concordance Index and Pearson’s correlation
  * Introduction to univariable vs. multivariable analysis for biomarkers
  
### Practical: Statistical Analysis

* Basic functionalities of PharmacoGx
* Case study – Univariable analysis of drug response using PharmacoGx 
* Plotting associations and interpretation of results in R

## Module 4: Pharmacogenomics for biomarker discovery – Advanced analysis

### Lecture:

* Meta-analysis
  * Introduction to “meta-analysis” in the context of inter-study investigations
  * Visualization and interpretation using forest plots
  * How to interpret meta-analysis statistics 
* Sub-group analysis
  * Identifying tissue-specific biomarkers with an example of cancer subtypes 
  * Other factors to consider (tumor grade, biological sex, etc)

### Practical: Statistical Analysis – Part 2

* Run meta-analysis on ~4 small PharmacoSets for a biomarker of interest for pan-cancer and cancer-specific data
* Generate and interpret forest plots with statistics
  
