---
title: "Create Test PSet"
author:
  - name: Jermiah J. Joseph
    email: jermiah.joseph@uhn.ca
    affiliation:
        - &pm Princess Margaret Cancer Centre
output: 
  BiocStyle::html_document:
    self_contained: yes
    toc: true
    toc_float: true
    toc_depth: 2
    code_folding: show
date: "`r doc_date()`"
package: "`r pkg_ver('CBWWorkshop2024')`"
vignette: >
  %\VignetteIndexEntry{Create Test PSet}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
---


<!-- Alias package links -->

```{r eval=TRUE, echo=FALSE}
PGx <- BiocStyle::Biocpkg("PharmacoGx")
CGx <- BiocStyle::Biocpkg("CoreGx")
TGx <- BiocStyle::Biocpkg("ToxicoGx")
RGx <- BiocStyle::Biocpkg("RadioGx")
Xv <- BiocStyle::Biocpkg("Xeva")
DT <- BiocStyle::CRANpkg("data.table")
SE <- BiocStyle::Biocpkg("SummarizedExperiment")
MAE <- BiocStyle::Biocpkg("MultiAssayExperiment")
```

```{r load_packages, eval=TRUE, echo=FALSE, include=FALSE}
suppressPackageStartupMessages({
    library(CBWWorkshop2024)
    library(PharmacoGx)
    library(data.table)
    library(ggplot2)
})

```


## Building A Drug Combination PharmacoSet

### `DataMapper` Class

The `DataMapper` is an abstract class included in `CoreGx` which defines
metadata specifying a "recipe" to go from rawdata files associated
with a specific experiment or publication to a curated and standardized data
format within an R `S4` class.

In `r CGx` we have implemented the `TREDataMapper` class, which encodes how
the rawdata from a drug combination experiment should be transformed
into its associated `TRE` object. A primary motivation for the creation of this
class was to document and standardize creation of `TRE` objects from the myriad
experimental designs available in drug combination experiments in the wild. It
is our hope that, for example, 3x3 drug combination experiments have enough in
common that we could theoretically create a `3by3TREDataMapper` which works
for (almost) all publications following this design.

As things stand right now, the `TREDataMapper` primarily acts as helper class
to explore the specifics of a drug combination experimental design. It allows
us to generate hypotheses about which identifiers are required to identify
observations in an assay. Below we will see how, along with the `guessMapping`
method, we can use the `TREDataMapper` to ensure we have mapped all data
from the NCI-ALMANAC drug combination experiment into a `TRE` object, ensuring
that all study observations and relevant metadata are captured in our
`TRE` and therefore in our `PharmacoSet`.

By curating and releasing such `PharmacoSet` objects, we ensure that our users
can access all information from the original publication via `r PGx` and
therefore easily ensure the reproducibility of the results as well as simplify
their comparison against other curated mono and combination therapy
`PharmacoSet`s object. In doing so, we help to improve the reproducibility and
consistency of _in vitro_ pharmacogenomic experiments and accelerate translation
of insights from the lab into the clinic.

### Using the `TREDataMapper` to Construct a `TreatmentResponseExperiment`

For this workshop we have included a subset of the NCI_ALMANAC_2017
`PharmacoSet`, curated from [@holbeckNationalCancerInstitute2017]. When approaching new
treament response experiment data, the first step is to merge it into a single,
long format table by attaching additional metadata for treatments, sample and
any other entities defined in the publication. We have done this already and
included it as the `NCI_ALMANAC_raw` dataset in the workshop package.

```{r nci_rawdata_echo, eval=FALSE, echo=TRUE}
data(NCI_ALMANAC_raw)
(head(NCI_ALMANAC_raw))
```

```{r nci_rawdata_eval, eval=TRUE, echo=FALSE}
data(NCI_ALMANAC_raw)
knitr::kable(head(NCI_ALMANAC_raw, 5))
```

Viewing the table, we see that each column represents a variable, and that
the number of rows in the table corresponds to the number of dose-response
observations measured in the `NCI-ALMANAC` study. To those familiar with
the R `tidyverse`, you may recognize that this dataset is in
[tidy format](https://r4ds.had.co.nz/tidy-data.html). It is also said be
in a "long" format because all observations are contained in a single column;
contrast this with a "wide" format where columns generally respresent samples
and values are observations are stored across both rows and columns
(_i.e._, as in a matrix or tensor).

In order for the helper methods of the `TREDataMapper` class to work, our
rawdata must first be in a single large, tidy `data.frame` or `data.table`.
Since these conditions are already met for the included example data, we can
create of `TREDataMapper` object and start exploring the to dimensionality of
the experiment in question!

```{r data_mapper_creation, eval=TRUE, echo=TRUE}
print(treDataMapper <- TREDataMapper(rawdata=NCI_ALMANAC_raw))
```

A `DataMapper` as a concept is simply a collection of metadata around our
raw data which defines how different columns map to different aspects of our
target `S4` class. For a `TRE`, similar to a `r SE`, we have the concept of
`rowData` which defines what is considered a treatment
(a specific drug combination, its doses and potential technical replicates) and
`colData` which defines what is considered a sample (a specific cell-line and
possible biological replicate identifier).

The `guessMapping` method can be used to test our hypotheses about which
information maps to treatments (rows), samples (columns) or assays
(experimental observations). This is done by defining a set of groupings
and querying the table to find other columns which map 1:1 with those groups.

For our first guess, lets assume there are no replicates and see if we are
able to map all of our rawdata columns using just treatment, treatment dose
and sample identifiers. Once we format our guess for rows, columns and
assays, we specify the subset parmeter to indicate we want to remove any
mapped columns before moving on to the next group. This prevents multimapping
a column to several groups and is generally recommended.


```{r data_mapper_incorrect_guess, eval=TRUE, echo=TRUE}
groups1 <- list(
    rowDataMap=c("treatment1id", "treatment2id", "treatment1dose",
        "treatment2dose"),
    colDataMap=c("sampleid"),
    assayMap=c("treatment1id", "treatment2id", "treatment1dose",
        "treatment2dose", "sampleid"
    )
)

(guess1 <- guessMapping(treDataMapper, groups=groups1, subset=TRUE))
```

Observing the results of our first guess, we can see that many column remain
unmapped. This likely indicates there is an additional piece of information
needed to uniquely identify the dimensions of our drug combination experiment.
One potential cause for this is inclusion of technical or biological
replicates.

Indeed, for the NCI-ALMANAC both of these kinds of replicates are present.

```{r debugging_incorrect_guess, eval=TRUE, echo=TRUE}
str(rawdata(treDataMapper), give.attr=FALSE)
```

Examining the columns in our data, we see we have already defined the both
`bio_rep`s and `tech_rep`s during raw data in preprocessing. We identified
biological replicates as observations of the sample treatment, dose and sample
combination across multiple dates. Technical replicates were the defined as any
additional rows not mapped to treatment, dose, sample and biological replicates.

For your own data in the wild, you will likely need to consult the original
publication and explore the data to differentiate these two types of replicates
and assign identifiers which make every observation within an experiment unique.

For our second guess we will include these columns, mapping biological
replicates to samples and technical replicates to treatments.

```{r data_mapper_correct_guess, eval=TRUE, echo=TRUE}
groups2 <- list(
    rowDataMap=c("treatment1id", "treatment2id", "treatment1dose",
        "treatment2dose", "tech_rep"),
    colDataMap=c("sampleid", "bio_rep"),
    assayMap=c("treatment1id", "treatment2id", "treatment1dose",
        "treatment2dose",  "sampleid", "bio_rep", "tech_rep")
)

(guess2 <- guessMapping(treDataMapper, groups=groups2, subset=TRUE))
```

Since we no longer have any unmapped columns, we know that we have found
valid dimension identifiers for our data. It is possible that some datasets
will have multiple valid dimension definitions, so further exploration may
be required if the first solution doesn't fit with your domain knowledge of the
experiment or with the design of the `TRE`, which stores data more efficiently
when only one dimension is large.

Lets assign our mapped columns back to the `TREDataMapper` so we can use
it to construct a `TreatmentResponseExperiment`.

For identified metadata, we need to assign a name to be used in the `metadata`
list of the createed `TRE`, other dimension maps can be assigned directly
to the slots of the `TREDataMapper`.

```{r updating_the_data_mapper, eval=TRUE, echo=TRUE}
metadataMap(treDataMapper) <- list(experiment_metadata=guess2$metadata$mapped_columns)
rowDataMap(treDataMapper) <- guess2$rowDataMap
colDataMap(treDataMapper) <- guess2$colDataMap
```

Unless we wish to have a single large assay in our `TRE`, we should further
subdivide our assay columns into sensible groups.

```{r splitting_up_the_assays1}
(guess2$assayMap)
```

For the NCI-ALMANAC we have decided to separate our treatment-response
measurements into the "response" assay. Any additional metadata which doesn't
map cleaning to rows or columns gets assigned to "assay_metadata". The publsihed
synergy score will go in the "profiles" assay, which generally stores metrics
computed from the "response" assay.

For our assay definitions we setup a nested list where each item defines an
assay in the `TRE`, with the "id_columns" containing identifiers needed to
uniquely map each row in that assay to its associated treatments and samples.
The "mapped_columns" item specifies the definition of the assay table.

For this case, all of our assays have the same identifier columns. However,
this is not a requirement and it may be more efficient to test further
subgroupings with our identifier columns to prevent uncessarily repeating
values in our `TRE`.

```{r splitting_up_the_assays2}
assaymap <- list(
    response=list(
        id_columns=guess2$assayMap$id_columns,
        mapped_columns=c("viability", "PERCENTGROWTH", "PERCENTGROWTHNOTZ",
            "EXPECTEDGROWTH", "TESTVALUE", "CONTROLVALUE", "TZVALUE")
    ),
    profiles=list(
        id_columns=guess2$assayMap$id_columns,
        mapped_columns=c("SCORE")
    ),
    assay_metadata=list(
        id_columns=guess2$assayMap$id_columns,
        mapped_columns=c("COMBODRUGSEQ", "STUDY", "TESTDATE", "PLATE",
            "SCREENER", "PREFIX2", "CONCINDEX1", "CONCINDEX2", "SAMPLE1",
            "SAMPLE2")
    )
)
assayMap(treDataMapper) <- assaymap
(treDataMapper)
```

Now that we have assigned all the necessary metadata to the `TREDataMapper`,
creating the object is as simple as calling `metaConstruct`, which combines
together the rawdata and the metadata into a constructor call for the
`TreatmentResponseExperiment`.

```{r metaconstruct_tre}
(NCI_ALMANAC_tre <- metaConstruct(treDataMapper))
```

### Computing Over a `TreatmentResponseExperiment`

So far we have seen how a `TRE` allows for flexible definitions of treatments
and samples to enable storage of dose-response observations across a range of
experimental designs. However, data storage is not the sole point of the `TRE`.
Once our data is curated and stored safely inside our object we are ready to
start answering interesting questions about the experiment.

To facilitate computation over a `TRE`, we have included the `aggregate` and
`endoaggregate` methods which allows arbitrary R functions to be applied over
groups of identifiers within an assay. For example, we may want to take averages
for all monotherapy viabilities over technical replicates. We can do so using
the `aggregate` method as follows:

```{r aggregate_tre}
(NCI_ALMANAC_tre |>
  subset(treatment2id == "") |>
  aggregate(
    assay="response",
    mean(viability),
    by=c("treatment1id", "treatment1dose", "sampleid", "bio_rep")
  ) ->
  mono_response_no_reps)
```

As you can see above, we select only our monotherapy response observations using
the `subset` method. Aggregate provides an argument for the `assay` we want
compute over, in this case the "response" assay. We also specify how we want our
groups defined using the `by` argument. Finally, we can specify one or more
functions to aggregate columns with. In this case, we are simply taking the
mean of viability within groups defined by `by`. We have excluded "tech_rep" for
the by argument which means that our summary will be computed over it.

The resulting table will contain the identifer columns specified to `by`,
as well as any columns which were summarized. By default, for
unnamed aggregations, `aggregate` tries to guess the column name. We can see in
the results it figured out that we are taking the mean of the viability column
and returned the results as "mean_viability".

You can also customize your resulting column names by naming the aggregation
arguments:

For example:
```{r aggregate_tre_named}
(NCI_ALMANAC_tre |>
  aggregate(
    subset=treatment2id == "",
    assay="response",
    my_viability=mean(viability),
    my_dose=mean(treatment1dose),
    by=c("treatment1id", "treatment1dose", "sampleid", "bio_rep")
  ) ->
  mono_response_custom_names)
```

Here we can see that the viability column is named "my_viability" and that
our dose column named "my_dose". We also use the `subset` parameter, which
was added to `aggregate` for convenience to filter to relevant assay rows
in a single function call.

For more complex operations, such as optimizing parameters to a non-linear
function (as is done for fitting the Hill curve model to dose-response data),
we have provided the `nthread` argument to parallelize computations when
we have additional compute resources available. In generaly, parallelized
computations with aggregate will use much more RAM than their serialized
counterparts. Since there is also overhead associated with dispatching parallel
threads for a computation, this feature should be used sparingly for only
complex and time consuming computations.

<!-- We will see examples of parallelized aggregation in subsequent sections. -->

The final feature of `aggregate` is allowing information be shared between
summarized columns to prevent the need for multiple parallelized calls On
our `TRE` object. For these cases, we set the `enlist` parameter to `FALSE` and
pass a block of code wrapped in `{}`. The final statment in the curly brackets
should return a named list, where each list item will become a column in the
returned table.

The utility of this option will also become apparent in subsequent sections.

### Combining with Omics Data into a `PharmacoSet`

The NCI-ALMANAC study did not conduct its own set of molecular profiling
experiments, since the NCI60 already profiled all cell-lines
available in the NCI-ALMANAC. We have already curated a `PharmacoSet`
for `NCI60`, so adding our molecular data to the NCI-ALMANAC `PharmacoSet`
requires no manual curation. We must simply subset the `NCI60` dataset
to those cell-lines present in NCI-ALMANAC.

Lets examine the included molecualr data:
```{r load_molecular_data}
data(NCI60_molecular_data)
(NCI60_molecular_data)
```

We can see the molecular data is available as `MultiAssayExperiment` containing
four `RangedSummarizedExperiment` objects. The molecular data types are:

  1. "rna" represents RNA expression microarray data.
  2. "mirna" represents microRNA expression data.
  3. "rnaseq.comp" contains RNA sequencing data.
  4. "rnaseq.iso" is RNA sequencing data specifically targeting transcript isoforms.

Once we have our `MultiAssayExperiment` and `TreatmentResponseExperiment`
objects ready, we next have to extract and curate sample and treatment
metadata tables which store a summary of metadata for available samples and
treatments in our `PharmacoSet` object.

For brevity, we have included preassembled annotation files for NCI-ALMANAC
since manual curation is beyong the scope of this workshop. When creating your
own `PharmacoSet`, we recommend assembling a set of standardized annotations
from sources like PubChem or ChEMBL for treatments and Cellosaurus for
cancer cell-lines and using these to standardize identifiers in your data
as well as adding links to additional information in external databases.

```{r load_sample_metadata_files1, eval=FALSE}
data(NCI_ALMANAC_sample_metadata)
head(NCI_ALMANAC_sample_metadata)
```

```{r load_sample_metadata_files2, echo=FALSE}
data(NCI_ALMANAC_sample_metadata)
knitr::kable(head(NCI_ALMANAC_sample_metadata))
```

The above table contains the sample metadata which will comprise the `@sample`
slot of our `PharmacoSet`. It contains a set of standardized identifiers
including `cellid` and `tissueid`, as well as cell-line identifiers from
the NCI-ALMANAC publication and some additional metadata we have added.

```{r load_treatment_metadata_files1, eval=FALSE}
data(NCI_ALMANAC_treatment_metadata)
head(NCI_ALMANAC_treatment_metadata)
```

```{r load_treatment_metadata_files2, echo=FALSE}
data(NCI_ALMANAC_treatment_metadata)
knitr::kable(head(NCI_ALMANAC_treatment_metadata))
```

Similar to the sample table, this one contains treatment metadata which will
be assigned to the `@treatment` slot of our `PharmacoSet`. It also contains
a set of standardized identifiers including `treatmentid`, structural
chemical identifiers, links to external databases as well as the drug identifiers
from the NCI-ALMANAC publication.

For the curation list, we extract our mappings from our standardized identifiers
to dataset specific identifiers and store those separately to ensure these
associations are not lost should the treatment or sample metadata tables
be modified.

```{r build_the_curation_list}
curation <- list(
  sample=NCI_ALMANAC_sample_metadata[, c("sampleid", "NCI_ALMANAC.sampleid")],
  treatment=NCI_ALMANAC_treatment_metadata[, c("treatmentid", "NCI_ALMANAC.treatmentid")],
  tissue=data.frame(
    NCI_ALMANAC_sample_metadata[, c("tissueid", "NCI_ALMANAC.disease")],
    rownames=NCI_ALMANAC_sample_metadata$sampleid
  )
)
```

With all of our pieces in place, we can finally construct a drug combination
`PharmacoSet` which can be used in subsequent analysis. Along with the recent
changes in the structure and nomenclature of a `PharmacoSet` comes the
introduction of the `PharmacoSet2` contructor. This version enforces new
requirements for the object such as a `MultiAssayExperiment` for the molecular
profiles slot and a `TreatmentResponseExperiment` for treatment response slot.
The original constructor is still included in the package for backwards
compatibility but will be retired after we finish updating all our existing
datasets on ORCESTRA.ca.

If you have an old style `PharmacoSet` but want to use the new package features,
simply call `updateObject` to return the dataset with an updated class
structure. This method will be available in all future releases to ensure
datasets can be updated and reused regardless of their age.

```{r create_our_drug_combo_pharmacoset}
(NCI_ALMANAC_2017 <- PharmacoSet2(
  name="NCI_ALMANAC_2017",
  molecularProfiles=NCI60_molecular_data,
  treatmentResponse=NCI_ALMANAC_tre,
  treatment=NCI_ALMANAC_treatment_metadata,
  sample=NCI_ALMANAC_sample_metadata,
  curation=curation
))
```

Finally we see our finished drug combination `PharmacoSet` and are ready
to use it to answer interesting research questions and discover the relationship
between molecular phenotypes and treatment response! 

