<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Module 1: Introduction to pharmacogenomics • CBWWorkshop2024</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Module 1: Introduction to pharmacogenomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CBWWorkshop2024</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9005</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/module1.html">Module 1: Introduction to pharmacogenomics</a></li>
    <li><a class="dropdown-item" href="../articles/module2.html">Module 2: Hands-on with pharmacogenomics data</a></li>
    <li><a class="dropdown-item" href="../articles/PGx_2024.html">Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Module 1: Introduction to pharmacogenomics</h1>
                        <h4 data-toc-skip class="author">Jermiah Joseph,
Nikta Feizi, Julia Nguyen</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a>
      </address>
                  
            <h4 data-toc-skip class="date">15 October 2024</h4>
      

      <div class="d-none name"><code>module1.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 class="unnumbered" id="lab-1-overview">Lab 1 Overview<a class="anchor" aria-label="anchor" href="#lab-1-overview"></a>
</h2>
<div class="section level3">
<h3 class="unnumbered" id="instructors-names-and-contact-information">Instructor(s) name(s) and contact
information<a class="anchor" aria-label="anchor" href="#instructors-names-and-contact-information"></a>
</h3>
<ul>
<li>Jermiah Joseph</li>
<li>Nikta Feizi</li>
<li>Julia Nguyen <a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 class="unnumbered" id="lab-description">Lab Description<a class="anchor" aria-label="anchor" href="#lab-description"></a>
</h3>
<!-- Lab Practical – Introduction to pharmacogenomics  (Jermiah Joseph, Nikta Feizi, Julia Nguyen)

Loading in expression matrices for various -omics data
Exploring feature extraction techniques to define biomarkers from (epi)genomic data -->
<div class="section level4">
<h4 class="unnumbered" id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h4>
<ul>
<li>Understand the data structure of a PharmacoSet</li>
<li>Learn how to access features and metadata from a PharmacoSet</li>
<li>Learn how to design linear multivariate predictors</li>
<li>Learn how to filter out outliers and missing values</li>
</ul>
</div>
<div class="section level4">
<h4 class="unnumbered" id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h4>
<ul>
<li>Describe the use cases for <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
in Pharmacogenomics</li>
<li>Understand the structure of the <code>CoreSet</code> and
<code>PharmacoSet</code> classes to facilitate their use in subsequent
analyses</li>
<li>Download/load a <code>PharmacoSet</code> using <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
or <a href="https://orcestra.ca" class="external-link">orcestra.ca</a>
</li>
<li>Subset and filter a <code>PharmacoSet</code> by samples and/or
treatments</li>
<li>Access the molecular features, dose-response and metadata contained
within the <code>PharmacoSet</code>
</li>
</ul>
<hr>
</div>
</div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<div class="section level3">
<h3 id="exploring-preclinical-datasets-for-pharmacogenomic-analysis">Exploring preclinical datasets for pharmacogenomic analysis<a class="anchor" aria-label="anchor" href="#exploring-preclinical-datasets-for-pharmacogenomic-analysis"></a>
</h3>
<p>*See list of available subsetted datasets from <a href="https://bhklab.github.io/CBWWorkshop2024/reference/index.html">Reference</a></p>
<div class="section level4">
<h4 id="molecular-profiles">Molecular profiles<a class="anchor" aria-label="anchor" href="#molecular-profiles"></a>
</h4>
<p>We will start with RNA-Seq data as a simple example.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_rnaseq</span><span class="op">)</span></span>
<span><span class="va">GDSC_rnaseq</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;    model_id model_name data_source   gene_id gene_symbol read_count  fpkm</span></span>
<span><span class="co">#&gt; 1 SIDM00794       A388      sanger SIDG00082       ABCC6         10  0.01</span></span>
<span><span class="co">#&gt; 2 SIDM00794       A388      sanger SIDG00106       ABCF3      20264 25.95</span></span>
<span><span class="co">#&gt; 3 SIDM00794       A388      sanger SIDG00108       ABCG2       1070  1.47</span></span>
<span><span class="co">#&gt; 4 SIDM00794       A388      sanger SIDG00148        ABI3          6  0.02</span></span>
<span><span class="co">#&gt; 5 SIDM00794       A388      sanger SIDG00177      ACADSB       1410  1.50</span></span>
<span><span class="co">#&gt; 6 SIDM00794       A388      sanger SIDG00198       ACER1         12  0.06</span></span></code></pre></div>
<p>Few key things to notice here: there are identifiers for the sample
(<code>model_id</code>, <code>model_name</code>), identifiers for the
gene (<code>gene_id</code>, <code>gene_symbol</code>), as well as two
expression values (<code>read_count</code>, <code>fpkm</code>).</p>
<p>When we create our expression matrices, we will select one sample
identifier, one feature (gene) identifier, and one expression value.</p>
</div>
<div class="section level4">
<h4 id="metadata-annotation-files">Metadata / annotation files<a class="anchor" aria-label="anchor" href="#metadata-annotation-files"></a>
</h4>
<p>When preparing for pharmacogenomic analysis, it is ideal to have
accompanying metadata for both the samples (cell lines) and the features
(genes).</p>
<p>We have made this data available through the package as well. We’ll
start with the gene annotations:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_gene_identifiers</span><span class="op">)</span></span>
<span><span class="va">GDSC_gene_identifiers</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;     gene_id cosmic_gene_symbol ensembl_gene_id entrez_id    hgnc_id hgnc_symbol</span></span>
<span><span class="co">#&gt; 1 SIDG00001               A1BG ENSG00000121410         1     HGNC:5        A1BG</span></span>
<span><span class="co">#&gt; 2 SIDG00002                    ENSG00000268895    503538 HGNC:37133    A1BG-AS1</span></span>
<span><span class="co">#&gt; 3 SIDG00003               A1CF ENSG00000148584     29974 HGNC:24086        A1CF</span></span>
<span><span class="co">#&gt; 4 SIDG00004                A2M ENSG00000175899         2     HGNC:7         A2M</span></span>
<span><span class="co">#&gt; 5 SIDG00005                    ENSG00000245105    144571 HGNC:27057     A2M-AS1</span></span>
<span><span class="co">#&gt; 6 SIDG00006              A2ML1 ENSG00000166535    144568 HGNC:23336       A2ML1</span></span>
<span><span class="co">#&gt;   refseq_id uniprot_id</span></span>
<span><span class="co">#&gt; 1 NM_130786     P04217</span></span>
<span><span class="co">#&gt; 2 NR_015380           </span></span>
<span><span class="co">#&gt; 3 NM_014576     Q9NQ94</span></span>
<span><span class="co">#&gt; 4 NM_000014     P01023</span></span>
<span><span class="co">#&gt; 5 NR_026971           </span></span>
<span><span class="co">#&gt; 6 NM_144670     A8K2U0</span></span></code></pre></div>
<p>The data above has been provided by GDSC and enables mapping across
various gene annotations. It is important to identify which gene
annotation maps to the RNA-Seq data and to check for completeness.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">gene_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">GDSC_gene_identifiers</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   TRUE </span></span>
<span><span class="co">#&gt; 135000</span></span>
<span></span>
<span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">GDSC_gene_identifiers</span><span class="op">$</span><span class="va">hgnc_symbol</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  FALSE   TRUE </span></span>
<span><span class="co">#&gt;   2100 132900</span></span>
<span></span>
<span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">GDSC_gene_identifiers</span><span class="op">$</span><span class="va">cosmic_gene_symbol</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt; 69600 65400</span></span></code></pre></div>
<p>This is a pretty obvious indicator to move forward with the gene_id
for downstream analysis.</p>
<p>Now we move over to the cell line annotations. There are a few
attributes made available, we first look to confirm the mapping id to
the RNA-Seq data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_model_list</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">GDSC_model_list</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "model_id"                                                     </span></span>
<span><span class="co">#&gt;  [2] "sample_id"                                                    </span></span>
<span><span class="co">#&gt;  [3] "patient_id"                                                   </span></span>
<span><span class="co">#&gt;  [4] "parent_id"                                                    </span></span>
<span><span class="co">#&gt;  [5] "model_name"                                                   </span></span>
<span><span class="co">#&gt;  [6] "synonyms"                                                     </span></span>
<span><span class="co">#&gt;  [7] "tissue"                                                       </span></span>
<span><span class="co">#&gt;  [8] "cancer_type"                                                  </span></span>
<span><span class="co">#&gt;  [9] "cancer_type_ncit_id"                                          </span></span>
<span><span class="co">#&gt; [10] "tissue_status"                                                </span></span>
<span><span class="co">#&gt; [11] "sample_site"                                                  </span></span>
<span><span class="co">#&gt; [12] "cancer_type_detail"                                           </span></span>
<span><span class="co">#&gt; [13] "model_type"                                                   </span></span>
<span><span class="co">#&gt; [14] "growth_properties"                                            </span></span>
<span><span class="co">#&gt; [15] "model_treatment"                                              </span></span>
<span><span class="co">#&gt; [16] "sampling_day"                                                 </span></span>
<span><span class="co">#&gt; [17] "sampling_month"                                               </span></span>
<span><span class="co">#&gt; [18] "sampling_year"                                                </span></span>
<span><span class="co">#&gt; [19] "doi"                                                          </span></span>
<span><span class="co">#&gt; [20] "pmed"                                                         </span></span>
<span><span class="co">#&gt; [21] "msi_status"                                                   </span></span>
<span><span class="co">#&gt; [22] "ploidy_snp6"                                                  </span></span>
<span><span class="co">#&gt; [23] "ploidy_wes"                                                   </span></span>
<span><span class="co">#&gt; [24] "ploidy_wgs"                                                   </span></span>
<span><span class="co">#&gt; [25] "mutational_burden"                                            </span></span>
<span><span class="co">#&gt; [26] "model_comments"                                               </span></span>
<span><span class="co">#&gt; [27] "model_relations_comment"                                      </span></span>
<span><span class="co">#&gt; [28] "COSMIC_ID"                                                    </span></span>
<span><span class="co">#&gt; [29] "BROAD_ID"                                                     </span></span>
<span><span class="co">#&gt; [30] "CCLE_ID"                                                      </span></span>
<span><span class="co">#&gt; [31] "RRID"                                                         </span></span>
<span><span class="co">#&gt; [32] "HCMI"                                                         </span></span>
<span><span class="co">#&gt; [33] "suppliers"                                                    </span></span>
<span><span class="co">#&gt; [34] "supplier"                                                     </span></span>
<span><span class="co">#&gt; [35] "cat_number"                                                   </span></span>
<span><span class="co">#&gt; [36] "species"                                                      </span></span>
<span><span class="co">#&gt; [37] "gender"                                                       </span></span>
<span><span class="co">#&gt; [38] "ethnicity"                                                    </span></span>
<span><span class="co">#&gt; [39] "age_at_sampling"                                              </span></span>
<span><span class="co">#&gt; [40] "family_history_of_cancer"                                     </span></span>
<span><span class="co">#&gt; [41] "prior_same_malignancy"                                        </span></span>
<span><span class="co">#&gt; [42] "prior_other_malignancy"                                       </span></span>
<span><span class="co">#&gt; [43] "smoking_status"                                               </span></span>
<span><span class="co">#&gt; [44] "alcohol_exposure_intensity"                                   </span></span>
<span><span class="co">#&gt; [45] "alcohol_consumption_per_week"                                 </span></span>
<span><span class="co">#&gt; [46] "history_diabetes"                                             </span></span>
<span><span class="co">#&gt; [47] "diabetes_treatment"                                           </span></span>
<span><span class="co">#&gt; [48] "colorectal_cancer_risk_factors"                               </span></span>
<span><span class="co">#&gt; [49] "patient_history_of_gastrointestinal_disorder"                 </span></span>
<span><span class="co">#&gt; [50] "gastrointestinal_disorder_diagnosed"                          </span></span>
<span><span class="co">#&gt; [51] "prior_synchronous_colon_rectal_tumours"                       </span></span>
<span><span class="co">#&gt; [52] "history_prior_colon_polyps"                                   </span></span>
<span><span class="co">#&gt; [53] "previous_current_barretts_oesophagus"                         </span></span>
<span><span class="co">#&gt; [54] "reflux_disease_prior"                                         </span></span>
<span><span class="co">#&gt; [55] "reflux_disease_treatment"                                     </span></span>
<span><span class="co">#&gt; [56] "h_pylori_infection"                                           </span></span>
<span><span class="co">#&gt; [57] "history_clinical_chronic_pancreatitis"                        </span></span>
<span><span class="co">#&gt; [58] "tumour_grade"                                                 </span></span>
<span><span class="co">#&gt; [59] "tnm_integrated"                                               </span></span>
<span><span class="co">#&gt; [60] "tnm_t"                                                        </span></span>
<span><span class="co">#&gt; [61] "tnm_n"                                                        </span></span>
<span><span class="co">#&gt; [62] "tnm_m"                                                        </span></span>
<span><span class="co">#&gt; [63] "clinical_staging"                                             </span></span>
<span><span class="co">#&gt; [64] "metastasis_at_diagnosis"                                      </span></span>
<span><span class="co">#&gt; [65] "metastatic_site_at_diagnosis"                                 </span></span>
<span><span class="co">#&gt; [66] "method_procurement"                                           </span></span>
<span><span class="co">#&gt; [67] "sample_treatment"                                             </span></span>
<span><span class="co">#&gt; [68] "sample_treatment_details"                                     </span></span>
<span><span class="co">#&gt; [69] "radiation_type"                                               </span></span>
<span><span class="co">#&gt; [70] "prior_therapy_outcome"                                        </span></span>
<span><span class="co">#&gt; [71] "tumour_regression_score"                                      </span></span>
<span><span class="co">#&gt; [72] "colon_polyps_present_at_time_of_tissue_collection"            </span></span>
<span><span class="co">#&gt; [73] "venous_invasion_present"                                      </span></span>
<span><span class="co">#&gt; [74] "venous_invasion_type"                                         </span></span>
<span><span class="co">#&gt; [75] "lymphatic_invasion_present"                                   </span></span>
<span><span class="co">#&gt; [76] "perineural_invasion_present"                                  </span></span>
<span><span class="co">#&gt; [77] "gastric_oesophageal_junction"                                 </span></span>
<span><span class="co">#&gt; [78] "barretts_goblet_cells_present"                                </span></span>
<span><span class="co">#&gt; [79] "oesophageal_columnar_metaplasia"                              </span></span>
<span><span class="co">#&gt; [80] "goblet_cells_oesophageal_columnar_mucosa"                     </span></span>
<span><span class="co">#&gt; [81] "degree_of_dysplasia_non_cancerous_oesophageal_columnar_mucosa"</span></span>
<span><span class="co">#&gt; [82] "peripancreatic_lymph_nodes_tested_surgery"                    </span></span>
<span><span class="co">#&gt; [83] "number_positive_peripancreatic_lymph_nodes"                   </span></span>
<span><span class="co">#&gt; [84] "mlh1_expression_by_ihc"                                       </span></span>
<span><span class="co">#&gt; [85] "mlh1_promoter_methylation_status"                             </span></span>
<span><span class="co">#&gt; [86] "msh2_expression_by_ihc"                                       </span></span>
<span><span class="co">#&gt; [87] "pms2_expression_by_ihc"                                       </span></span>
<span><span class="co">#&gt; [88] "msh6_expression_by_ihc"                                       </span></span>
<span><span class="co">#&gt; [89] "braf_mutation_identified"                                     </span></span>
<span><span class="co">#&gt; [90] "braf_expression_by_ihc"                                       </span></span>
<span><span class="co">#&gt; [91] "pik3ca_mutation_identified"                                   </span></span>
<span><span class="co">#&gt; [92] "pten_expression_by_ihc"                                       </span></span>
<span><span class="co">#&gt; [93] "pten_mutation_identified"                                     </span></span>
<span><span class="co">#&gt; [94] "kras_mutation_identified"                                     </span></span>
<span><span class="co">#&gt; [95] "mismatch_repair_status"                                       </span></span>
<span><span class="co">#&gt; [96] "preoperative_ce_alevel"                                       </span></span>
<span><span class="co">#&gt; [97] "crispr_ko_data"</span></span>
<span></span>
<span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">model_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">GDSC_model_list</span><span class="op">$</span><span class="va">model_id</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   TRUE </span></span>
<span><span class="co">#&gt; 135000</span></span></code></pre></div>
<p>Below are some examples of other available variables that may be of
interest for downstream analysis.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GDSC_model_list</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'model_id'</span>, <span class="st">'model_name'</span>, <span class="st">'tissue'</span>, <span class="st">'ploidy_wes'</span>, <span class="st">'mutational_burden'</span>, <span class="st">'gender'</span>, <span class="st">'ethnicity'</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;    model_id model_name                      tissue ploidy_wes mutational_burden</span></span>
<span><span class="co">#&gt; 1 SIDM01774      PK-59                    Pancreas   3.510751             24.79</span></span>
<span><span class="co">#&gt; 2 SIDM00192   SNU-1033             Large Intestine   2.780367             23.29</span></span>
<span><span class="co">#&gt; 3 SIDM01447    SNU-466      Central Nervous System   2.054101             20.58</span></span>
<span><span class="co">#&gt; 4 SIDM01554  IST-MES-2                        Lung   1.851007             22.92</span></span>
<span><span class="co">#&gt; 5 SIDM01689     MUTZ-5 Haematopoietic and Lymphoid   1.941110             28.76</span></span>
<span><span class="co">#&gt; 6 SIDM01460      TM-31      Central Nervous System   2.885529             25.89</span></span>
<span><span class="co">#&gt;    gender  ethnicity</span></span>
<span><span class="co">#&gt; 1 Unknown    Unknown</span></span>
<span><span class="co">#&gt; 2  Female East Asian</span></span>
<span><span class="co">#&gt; 3    Male    Unknown</span></span>
<span><span class="co">#&gt; 4    Male      White</span></span>
<span><span class="co">#&gt; 5    Male    Unknown</span></span>
<span><span class="co">#&gt; 6  Female East Asian</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="drug-response-data">Drug response data<a class="anchor" aria-label="anchor" href="#drug-response-data"></a>
</h4>
<p>Finally, we’ll load in the corresponding drug response data for these
cell lines.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_drug_response</span><span class="op">)</span></span>
<span><span class="va">GDSC_drug_response</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   DATASET NLME_RESULT_ID NLME_CURVE_ID COSMIC_ID CELL_LINE_NAME SANGER_MODEL_ID</span></span>
<span><span class="co">#&gt; 1   GDSC2            343      15946320    683667         PFSK-1       SIDM01132</span></span>
<span><span class="co">#&gt; 2   GDSC2            343      15946560    684052           A673       SIDM00848</span></span>
<span><span class="co">#&gt; 3   GDSC2            343      15946840    684057            ES5       SIDM00263</span></span>
<span><span class="co">#&gt; 4   GDSC2            343      15947099    684059            ES7       SIDM00269</span></span>
<span><span class="co">#&gt; 5   GDSC2            343      15947381    684062          EW-11       SIDM00203</span></span>
<span><span class="co">#&gt; 6   GDSC2            343      15947663    684072        SK-ES-1       SIDM01111</span></span>
<span><span class="co">#&gt;      TCGA_DESC DRUG_ID DRUG_NAME PUTATIVE_TARGET     PATHWAY_NAME COMPANY_ID</span></span>
<span><span class="co">#&gt; 1           MB    1017  Olaparib    PARP1, PARP2 Genome integrity       1046</span></span>
<span><span class="co">#&gt; 2 UNCLASSIFIED    1017  Olaparib    PARP1, PARP2 Genome integrity       1046</span></span>
<span><span class="co">#&gt; 3 UNCLASSIFIED    1017  Olaparib    PARP1, PARP2 Genome integrity       1046</span></span>
<span><span class="co">#&gt; 4 UNCLASSIFIED    1017  Olaparib    PARP1, PARP2 Genome integrity       1046</span></span>
<span><span class="co">#&gt; 5 UNCLASSIFIED    1017  Olaparib    PARP1, PARP2 Genome integrity       1046</span></span>
<span><span class="co">#&gt; 6 UNCLASSIFIED    1017  Olaparib    PARP1, PARP2 Genome integrity       1046</span></span>
<span><span class="co">#&gt;   WEBRELEASE MIN_CONC MAX_CONC  LN_IC50      AUC     RMSE   Z_SCORE</span></span>
<span><span class="co">#&gt; 1          Y 0.010005       10 4.488810 0.974081 0.072391  0.201882</span></span>
<span><span class="co">#&gt; 2          Y 0.010005       10 1.782152 0.842679 0.068257 -1.881795</span></span>
<span><span class="co">#&gt; 3          Y 0.010005       10 2.116072 0.869909 0.070087 -1.624732</span></span>
<span><span class="co">#&gt; 4          Y 0.010005       10 1.685857 0.834608 0.092726 -1.955925</span></span>
<span><span class="co">#&gt; 5          Y 0.010005       10 2.078938 0.844879 0.114103 -1.653318</span></span>
<span><span class="co">#&gt; 6          Y 0.010005       10 0.592900 0.727416 0.081839 -2.797320</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">model_id</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">GDSC_drug_response</span><span class="op">$</span><span class="va">SANGER_MODEL_ID</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FALSE  TRUE </span></span>
<span><span class="co">#&gt;    10    90</span></span></code></pre></div>
<p>We can use <code>SANGER_MODEL_ID</code> to map back to our RNA-Seq
data. <code>DRUG_NAME</code> will be used as the identifier for the
treatment. We also have both the IC50 (<code>LN_IC50</code>) and the AUC
(<code>AUC</code>) values for each cell-drug pair.</p>
<p>Notice that some of the cell lines do not have drug response data.
These will need to be filtered before downstream analysis.</p>
</div>
</div>
<div class="section level3">
<h3 id="exploring-other-multi-omic-profiles">Exploring other multi-omic profiles<a class="anchor" aria-label="anchor" href="#exploring-other-multi-omic-profiles"></a>
</h3>
<p>We have prepared a variety of other molecular profiles from both GDSC
and CCLE. We look through a few more examples below to better understand
these data types.</p>
<div class="section level4">
<h4 id="driver-mutations">Driver mutations<a class="anchor" aria-label="anchor" href="#driver-mutations"></a>
</h4>
<p>Load in the driver mutations data from GDSC:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_drivers</span><span class="op">)</span></span>
<span><span class="va">GDSC_drivers</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;     gene_id gene_symbol  model_id protein_mutation             rna_mutation</span></span>
<span><span class="co">#&gt; 1 SIDG27130       RGPD3 SIDM02101       p.N241fs*6  r.809_819delAAUCUUAUGCU</span></span>
<span><span class="co">#&gt; 2 SIDG08129        ESR1 SIDM02095       p.L15fs*69        r.412_416delACUGC</span></span>
<span><span class="co">#&gt; 3 SIDG03559        CBLC SIDM02090      p.Q419fs*81          r.1295_1296insc</span></span>
<span><span class="co">#&gt; 4 SIDG02114        BAP1 SIDM02090           p.C91Y                 r.402g&gt;a</span></span>
<span><span class="co">#&gt; 5 SIDG02114        BAP1 SIDM02090           p.N78S                 r.363a&gt;g</span></span>
<span><span class="co">#&gt; 6 SIDG36265        SPEN SIDM02089      p.R753fs*53 r.2618_2627delAGGAGGCUUU</span></span>
<span><span class="co">#&gt;              cdna_mutation cancer_driver cancer_predisposition_variant</span></span>
<span><span class="co">#&gt; 1  c.721_731delAATCTTATGCT          True                         False</span></span>
<span><span class="co">#&gt; 2          c.42_46delACTGC          True                         False</span></span>
<span><span class="co">#&gt; 3          c.1253_1254insC          True                         False</span></span>
<span><span class="co">#&gt; 4                 c.272G&gt;A          True                         False</span></span>
<span><span class="co">#&gt; 5                 c.233A&gt;G          True                         False</span></span>
<span><span class="co">#&gt; 6 c.2257_2266delAGGAGGCTTT          True                         False</span></span>
<span><span class="co">#&gt;       effect    vaf coding source            model_name</span></span>
<span><span class="co">#&gt; 1 frameshift 0.2319   True Sanger Mesobank_CellLine-53T</span></span>
<span><span class="co">#&gt; 2 frameshift 0.4259   True Sanger  Mesobank_CellLine-26</span></span>
<span><span class="co">#&gt; 3 frameshift 0.5217   True Sanger Mesobank_CellLine-50T</span></span>
<span><span class="co">#&gt; 4   missense 0.5217   True Sanger Mesobank_CellLine-50T</span></span>
<span><span class="co">#&gt; 5   missense 0.4595   True Sanger Mesobank_CellLine-50T</span></span>
<span><span class="co">#&gt; 6 frameshift 0.6333   True Sanger  Mesobank_CellLine-45</span></span></code></pre></div>
<p>Notice that this data is not a continuous expression like the
RNA-Seq. This data will have to be further processed before it can be
used to predict response.</p>
</div>
<div class="section level4">
<h4 id="methylation">Methylation<a class="anchor" aria-label="anchor" href="#methylation"></a>
</h4>
<p>Load in the methylation matrix from GDSC:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_methylation</span><span class="op">)</span></span>
<span><span class="va">GDSC_methylation</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                          X8359018054_R03C01 X8359018053_R04C02</span></span>
<span><span class="co">#&gt; chr1:1051178-1052445              0.3733729          0.4144962</span></span>
<span><span class="co">#&gt; chr1:109824313-109824526          0.4644322          0.5959816</span></span>
<span><span class="co">#&gt; chr1:109825710-109826207          0.1411910          0.1770613</span></span>
<span><span class="co">#&gt; chr1:110008962-110010124          0.4143231          0.5317602</span></span>
<span><span class="co">#&gt; chr1:110527248-110528026          0.3022454          0.1882153</span></span>
<span><span class="co">#&gt;                          X8221932075_R03C02 X8221924165_R04C02</span></span>
<span><span class="co">#&gt; chr1:1051178-1052445              0.2892333          0.3023764</span></span>
<span><span class="co">#&gt; chr1:109824313-109824526          0.5048080          0.4594520</span></span>
<span><span class="co">#&gt; chr1:109825710-109826207          0.1668371          0.1698714</span></span>
<span><span class="co">#&gt; chr1:110008962-110010124          0.5259936          0.4551888</span></span>
<span><span class="co">#&gt; chr1:110527248-110528026          0.2679353          0.2989812</span></span>
<span><span class="co">#&gt;                          X7970368131_R04C02</span></span>
<span><span class="co">#&gt; chr1:1051178-1052445              0.3880624</span></span>
<span><span class="co">#&gt; chr1:109824313-109824526          0.5114854</span></span>
<span><span class="co">#&gt; chr1:109825710-109826207          0.1706867</span></span>
<span><span class="co">#&gt; chr1:110008962-110010124          0.4740822</span></span>
<span><span class="co">#&gt; chr1:110527248-110528026          0.2603969</span></span></code></pre></div>
<p>This data has already been processed into a matrix. Notice though
that the sample names are not present, instead there is the array ID and
position. We can use the provided annotation file to map back to the
sample names</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GDSC_methylation_model_list</span><span class="op">)</span></span>
<span><span class="va">GDSC_methylation_model_list</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Sample_Name Sample_Well Sample_Plate Sample_Group Pool_ID Sentrix_ID</span></span>
<span><span class="co">#&gt; 1       HL-60         A06     SMET0001           NA      NA 5684819030</span></span>
<span><span class="co">#&gt; 2      IGR-37         B06     SMET0001           NA      NA 5684819030</span></span>
<span><span class="co">#&gt; 3      WM793B         C07     SMET0001           NA      NA 5723654013</span></span>
<span><span class="co">#&gt; 4       IGR39         C08     SMET0001           NA      NA 5723654013</span></span>
<span><span class="co">#&gt; 5      SW-480         D09     SMET0001           NA      NA 5723654015</span></span>
<span><span class="co">#&gt; 6         C32         E09     SMET0001           NA      NA 5723654015</span></span>
<span><span class="co">#&gt;   Sentrix_Position   Investigator Project                             Tissue</span></span>
<span><span class="co">#&gt; 1           R05C01 Catia Moutinio    &lt;NA&gt; HAEMATOPOIETIC AND LYMPHOID TISSUE</span></span>
<span><span class="co">#&gt; 2           R06C01   Javi Carmona    &lt;NA&gt;                               SKIN</span></span>
<span><span class="co">#&gt; 3           R03C01   Javi Carmona    &lt;NA&gt;                               SKIN</span></span>
<span><span class="co">#&gt; 4           R05C02   Javi Carmona    &lt;NA&gt;                               SKIN</span></span>
<span><span class="co">#&gt; 5           R02C02   Javi Carmona    &lt;NA&gt;                    LARGE INTESTINE</span></span>
<span><span class="co">#&gt; 6           R03C02 Catia Moutinio    &lt;NA&gt;                               SKIN</span></span>
<span><span class="co">#&gt;                     Type EBV Cell_Line Wildtype Normal     Coment  Scan_Date</span></span>
<span><span class="co">#&gt; 1 ACUTE MYELOID LEUKEMIA  No       YES      Yes     No       &lt;NA&gt; 2011-02-12</span></span>
<span><span class="co">#&gt; 2               MELANOMA  No       YES      Yes     No Metastasis 2011-02-12</span></span>
<span><span class="co">#&gt; 3               MELANOMA  No       YES      Yes     No Metastasis 2011-02-12</span></span>
<span><span class="co">#&gt; 4               MELANOMA  No       YES      Yes     No    Primary 2011-02-12</span></span>
<span><span class="co">#&gt; 5         ADENOCARCINOMA  No       YES      Yes     No    Primary 2011-02-12</span></span>
<span><span class="co">#&gt; 6               MELANOMA  No       YES      Yes     No       &lt;NA&gt; 2011-02-12</span></span>
<span><span class="co">#&gt;   GDSC1                   GDSC2 cosmic_id</span></span>
<span><span class="co">#&gt; 1 blood acute_myeloid_leukaemia    905938</span></span>
<span><span class="co">#&gt; 2  skin                melanoma   1240153</span></span>
<span><span class="co">#&gt; 3  skin                melanoma   1299081</span></span>
<span><span class="co">#&gt; 4  skin                melanoma   1298148</span></span>
<span><span class="co">#&gt; 5  &lt;NA&gt;                    &lt;NA&gt;        NA</span></span>
<span><span class="co">#&gt; 6  skin                melanoma    906830</span></span>
<span></span>
<span><span class="va">GDSC_methylation_model_list</span><span class="op">$</span><span class="va">sampleid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="va">GDSC_methylation_model_list</span><span class="op">$</span><span class="va">Sentrix_ID</span>, <span class="st">"_"</span>, <span class="va">GDSC_methylation_model_list</span><span class="op">$</span><span class="va">Sentrix_Position</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">GDSC_methylation</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">GDSC_methylation_model_list</span><span class="op">$</span><span class="va">sampleid</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; TRUE </span></span>
<span><span class="co">#&gt;  100</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">GDSC_methylation</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">GDSC_methylation_model_list</span><span class="op">$</span><span class="va">Sample_Name</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">GDSC_methylation</span><span class="op">)</span>, <span class="va">GDSC_methylation_model_list</span><span class="op">$</span><span class="va">sampleid</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">GDSC_methylation</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                               A673       RT4   8-MG-BA  U-118-MG CHAGO-K-1</span></span>
<span><span class="co">#&gt; chr1:1051178-1052445     0.3733729 0.4144962 0.2892333 0.3023764 0.3880624</span></span>
<span><span class="co">#&gt; chr1:109824313-109824526 0.4644322 0.5959816 0.5048080 0.4594520 0.5114854</span></span>
<span><span class="co">#&gt; chr1:109825710-109826207 0.1411910 0.1770613 0.1668371 0.1698714 0.1706867</span></span>
<span><span class="co">#&gt; chr1:110008962-110010124 0.4143231 0.5317602 0.5259936 0.4551888 0.4740822</span></span>
<span><span class="co">#&gt; chr1:110527248-110528026 0.3022454 0.1882153 0.2679353 0.2989812 0.2603969</span></span></code></pre></div>
<p>We have provided a few other subsetted datasets. A full list is
available from <a href="https://bhklab.github.io/CBWWorkshop2024/reference/index.html">Reference</a></p>
<p>We encourage independent exploration of these datasets.</p>
</div>
</div>
<div class="section level3">
<h3 id="creating-expression-matrices-for-pharmacogenomic-analysis">Creating expression matrices for pharmacogenomic analysis<a class="anchor" aria-label="anchor" href="#creating-expression-matrices-for-pharmacogenomic-analysis"></a>
</h3>
<p>To facilitate downstream pharmacogenomic analysis, we want to create
an expression matrix such that: * Features are the rows * Samples are
the columns * Feature expression as the individual values</p>
<p>Below, we show a example of such matrix using dummy data</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dummy_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">replicate</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Sample"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">dummy_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Feature"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">dummy_data</span></span>
<span><span class="co">#&gt;               Sample1    Sample2    Sample3     Sample4    Sample5</span></span>
<span><span class="co">#&gt; Feature1 -1.400043517  1.1484116 -0.5536994 -1.86301149  0.4681544</span></span>
<span><span class="co">#&gt; Feature2  0.255317055 -1.8218177  0.6289820 -0.52201251  0.3629513</span></span>
<span><span class="co">#&gt; Feature3 -2.437263611 -0.2473253  2.0650249 -0.05260191 -1.3045435</span></span>
<span><span class="co">#&gt; Feature4 -0.005571287 -0.2441996 -1.6309894  0.54299634  0.7377763</span></span>
<span><span class="co">#&gt; Feature5  0.621552721 -0.2827054  0.5124269 -0.91407483  1.8885049</span></span></code></pre></div>
<p>Let’s revisit the RNA-Seq example. The data is currently in a long
format (i.e. there is one row for each sample-feature observation).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GDSC_rnaseq</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;    model_id model_name data_source   gene_id gene_symbol read_count  fpkm</span></span>
<span><span class="co">#&gt; 1 SIDM00794       A388      sanger SIDG00082       ABCC6         10  0.01</span></span>
<span><span class="co">#&gt; 2 SIDM00794       A388      sanger SIDG00106       ABCF3      20264 25.95</span></span>
<span><span class="co">#&gt; 3 SIDM00794       A388      sanger SIDG00108       ABCG2       1070  1.47</span></span>
<span><span class="co">#&gt; 4 SIDM00794       A388      sanger SIDG00148        ABI3          6  0.02</span></span>
<span><span class="co">#&gt; 5 SIDM00794       A388      sanger SIDG00177      ACADSB       1410  1.50</span></span>
<span><span class="co">#&gt; 6 SIDM00794       A388      sanger SIDG00198       ACER1         12  0.06</span></span>
<span><span class="va">GDSC_rnaseq</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 135000      7</span></span>
<span></span>
<span><span class="co"># number of cell line samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">model_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 100</span></span>
<span></span>
<span><span class="co"># number of genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">GDSC_rnaseq</span><span class="op">$</span><span class="va">gene_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1350</span></span></code></pre></div>
<p>We want to convert this into a wide format such that each row is a
gene, each column is a sample, and the values are the gene
expression.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/cast.html" class="external-link">dcast</a></span><span class="op">(</span><span class="va">GDSC_rnaseq</span>, <span class="va">gene_id</span> <span class="op">~</span> <span class="va">model_name</span>, value.var <span class="op">=</span> <span class="st">"fpkm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">expr</span><span class="op">$</span><span class="va">gene_id</span></span>
<span><span class="va">expr</span><span class="op">$</span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="va">expr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt;            A388  A427 BB65-RCC Becker BICR78 C-33-A Ca-Ski Ca9-22 CCK-81 CHL-1</span></span>
<span><span class="co">#&gt; SIDG00082  0.01  0.29     0.04   0.15   0.03   0.01   0.01   0.01   0.94  1.23</span></span>
<span><span class="co">#&gt; SIDG00106 25.95 16.45     8.19  13.88  14.04  10.99  16.42   9.04   9.37 14.86</span></span>
<span><span class="co">#&gt; SIDG00108  1.47  0.52     0.02   0.03   1.02   0.01   0.27   0.09   0.01  0.04</span></span>
<span><span class="co">#&gt; SIDG00148  0.02  0.02     0.35   0.01   0.00   0.02   0.00   0.10   0.00  0.00</span></span>
<span><span class="co">#&gt; SIDG00177  1.50  6.18     3.07   3.40   1.97   7.21   1.01   1.83   5.56 11.63</span></span>
<span></span>
<span><span class="va">expr</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1350  100</span></span></code></pre></div>
<p>Notice that we have the 1350 genes as the rows and the 100 cell lines
as the columns</p>
</div>
<div class="section level3">
<h3 id="feature-extraction-techniques-to-define-biomarkers">Feature extraction techniques to define biomarkers<a class="anchor" aria-label="anchor" href="#feature-extraction-techniques-to-define-biomarkers"></a>
</h3>
<p>While using the continuous expression of single features is a
convenient method for quantifying biomarkers, there are cases when other
techniques are needed and/or more appropriate.</p>
<div class="section level4">
<h4 id="binarization">Binarization<a class="anchor" aria-label="anchor" href="#binarization"></a>
</h4>
<p>Recall that the driver mutations data was not presented as a
continuous numeric values. One method to prepare this data is to
binarize the mutation status.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GDSC_drivers</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;     gene_id gene_symbol  model_id protein_mutation             rna_mutation</span></span>
<span><span class="co">#&gt; 1 SIDG27130       RGPD3 SIDM02101       p.N241fs*6  r.809_819delAAUCUUAUGCU</span></span>
<span><span class="co">#&gt; 2 SIDG08129        ESR1 SIDM02095       p.L15fs*69        r.412_416delACUGC</span></span>
<span><span class="co">#&gt; 3 SIDG03559        CBLC SIDM02090      p.Q419fs*81          r.1295_1296insc</span></span>
<span><span class="co">#&gt; 4 SIDG02114        BAP1 SIDM02090           p.C91Y                 r.402g&gt;a</span></span>
<span><span class="co">#&gt; 5 SIDG02114        BAP1 SIDM02090           p.N78S                 r.363a&gt;g</span></span>
<span><span class="co">#&gt; 6 SIDG36265        SPEN SIDM02089      p.R753fs*53 r.2618_2627delAGGAGGCUUU</span></span>
<span><span class="co">#&gt;              cdna_mutation cancer_driver cancer_predisposition_variant</span></span>
<span><span class="co">#&gt; 1  c.721_731delAATCTTATGCT          True                         False</span></span>
<span><span class="co">#&gt; 2          c.42_46delACTGC          True                         False</span></span>
<span><span class="co">#&gt; 3          c.1253_1254insC          True                         False</span></span>
<span><span class="co">#&gt; 4                 c.272G&gt;A          True                         False</span></span>
<span><span class="co">#&gt; 5                 c.233A&gt;G          True                         False</span></span>
<span><span class="co">#&gt; 6 c.2257_2266delAGGAGGCTTT          True                         False</span></span>
<span><span class="co">#&gt;       effect    vaf coding source            model_name</span></span>
<span><span class="co">#&gt; 1 frameshift 0.2319   True Sanger Mesobank_CellLine-53T</span></span>
<span><span class="co">#&gt; 2 frameshift 0.4259   True Sanger  Mesobank_CellLine-26</span></span>
<span><span class="co">#&gt; 3 frameshift 0.5217   True Sanger Mesobank_CellLine-50T</span></span>
<span><span class="co">#&gt; 4   missense 0.5217   True Sanger Mesobank_CellLine-50T</span></span>
<span><span class="co">#&gt; 5   missense 0.4595   True Sanger Mesobank_CellLine-50T</span></span>
<span><span class="co">#&gt; 6 frameshift 0.6333   True Sanger  Mesobank_CellLine-45</span></span></code></pre></div>
<p>Looking at the first row, we can see that there is a mutation on the
<code>RGPD3</code> gene in the <code>SIDM02101</code> cell line model.
We would represent such mutation events wtih <code>1</code>.</p>
<p>The code below again casts the long data frame into a wide format.
This time we specify an aggregate function <code><a href="https://rdrr.io/r/base/length.html" class="external-link">length()</a></code>, which
returns the number of rows (mutation events) for each gene-cell line
pair. This was done by passing the option
<code>fun.aggregate = length</code></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/cast.html" class="external-link">dcast</a></span><span class="op">(</span><span class="va">GDSC_drivers</span>, <span class="va">gene_symbol</span> <span class="op">~</span> <span class="va">model_id</span>, value.var <span class="op">=</span> <span class="st">"cdna_mutation"</span>, fun.aggregate <span class="op">=</span> <span class="va">length</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">expr</span><span class="op">$</span><span class="va">gene_symbol</span></span>
<span><span class="va">expr</span><span class="op">$</span><span class="va">gene_symbol</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="va">expr</span><span class="op">[</span><span class="st">"RGPD3"</span>, <span class="st">"SIDM02101"</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span></span>
<span><span class="va">expr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt;       SIDM00001 SIDM00002 SIDM00003 SIDM00006 SIDM00007 SIDM00008 SIDM00009</span></span>
<span><span class="co">#&gt; ABCB1         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ABI1          0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ABL1          0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ABL2          0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt; ACVR1         0         0         0         0         0         0         0</span></span>
<span><span class="co">#&gt;       SIDM00011 SIDM00013 SIDM00014</span></span>
<span><span class="co">#&gt; ABCB1         0         0         0</span></span>
<span><span class="co">#&gt; ABI1          0         0         0</span></span>
<span><span class="co">#&gt; ABL1          0         0         0</span></span>
<span><span class="co">#&gt; ABL2          0         1         0</span></span>
<span><span class="co">#&gt; ACVR1         0         0         0</span></span></code></pre></div>
<p>There was one mutation event on the <code>RGPD3</code> gene in the
<code>SIDM02101</code> cell line model, hence the value of this
combination is <code>1</code>.</p>
<p>Mutation events are relatively sparse, so we see <code>0</code> for
the majority of the matrix.</p>
</div>
<div class="section level4">
<h4 id="signature-extraction">Signature extraction<a class="anchor" aria-label="anchor" href="#signature-extraction"></a>
</h4>
<p>There are cases when individual features have low predictive power,
but when combined become much more informative of drug response.</p>
<p>Let’s revisit our methylation data. Recall that each row is a CpG
site. There are 1000 CpG sites.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GDSC_methylation</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                               A673       RT4   8-MG-BA  U-118-MG CHAGO-K-1</span></span>
<span><span class="co">#&gt; chr1:1051178-1052445     0.3733729 0.4144962 0.2892333 0.3023764 0.3880624</span></span>
<span><span class="co">#&gt; chr1:109824313-109824526 0.4644322 0.5959816 0.5048080 0.4594520 0.5114854</span></span>
<span><span class="co">#&gt; chr1:109825710-109826207 0.1411910 0.1770613 0.1668371 0.1698714 0.1706867</span></span>
<span><span class="co">#&gt; chr1:110008962-110010124 0.4143231 0.5317602 0.5259936 0.4551888 0.4740822</span></span>
<span><span class="co">#&gt; chr1:110527248-110528026 0.3022454 0.1882153 0.2679353 0.2989812 0.2603969</span></span>
<span></span>
<span><span class="va">GDSC_methylation</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1000  100</span></span></code></pre></div>
<p>Signatures refer to combinations of features that form some pattern
with biological relevance. For example, you may choose to define a
signature X to represent CpG sites located on promoters of genes
involved in pathway Y).</p>
<p>For simplicity, we can define some arbituary signatures from our CpG
sites.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span> </span>
<span><span class="va">signatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>CpG <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">GDSC_methylation</span><span class="op">)</span>, </span>
<span>                         Signature <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/gl.html" class="external-link">gl</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">100</span>, length <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">signatures</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                        CpG Signature</span></span>
<span><span class="co">#&gt; 1     chr1:1051178-1052445         5</span></span>
<span><span class="co">#&gt; 2 chr1:109824313-109824526         5</span></span>
<span><span class="co">#&gt; 3 chr1:109825710-109826207         2</span></span>
<span><span class="co">#&gt; 4 chr1:110008962-110010124         6</span></span>
<span><span class="co">#&gt; 5 chr1:110527248-110528026         2</span></span>
<span><span class="co">#&gt; 6 chr1:114354373-114355300        10</span></span></code></pre></div>
<p>Each of the 1000 CpG sites was randomly assigned to one of 10
signatures.</p>
<p>Next we want to quantify the signature for each cell line. Again for
simplicity, we will sum the beta values for across each CpG for each
signature.</p>
<p>We now have a new expression matrix, this time of the 10 defined
signatures for each of our cell lines.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jermiah Joseph, Julia Ngyugen.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
