<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Module 2: Hands-on with pharmacogenomics data • CBWWorkshop2024</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Module 2: Hands-on with pharmacogenomics data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CBWWorkshop2024</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9025</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Introduction</h6></li>
    <li><a class="dropdown-item" href="../articles/PGx_2024.html">Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance</a></li>
    <li><a class="dropdown-item" href="../articles/Workshop_Outline.html">Workshop Outline and Objectives</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Module Labs</h6></li>
    <li><a class="dropdown-item" href="../articles/Module1.html">Module 1: Introduction to pharmacogenomics</a></li>
    <li><a class="dropdown-item" href="../articles/Module2.html">Module 2: Hands-on with pharmacogenomics data</a></li>
    <li><a class="dropdown-item" href="../articles/Module3.html">Module 3: Pharmacogenomics for biomarker discovery – Basic analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Module4.html">Module 4: Pharmacogenomics for biomarker discovery - Advanced analysis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/bhklab/CBWWorkshop2024"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Module 2: Hands-on with pharmacogenomics data</h1>
                        <h4 data-toc-skip class="author">Jermiah J.
Joseph</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
      </address>
                              <h4 data-toc-skip class="author">Julia
Nguyen</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/bhklab/CBWWorkshop2024/blob/main/vignettes/Module2.Rmd" class="external-link"><code>vignettes/Module2.Rmd</code></a></small>
      <div class="d-none name"><code>Module2.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 class="unnumbered" id="lab-2-overview">Lab 2 Overview<a class="anchor" aria-label="anchor" href="#lab-2-overview"></a>
</h2>
<div class="section level3">
<h3 class="unnumbered" id="instructors-names-and-contact-information">Instructor(s) name(s) and contact
information<a class="anchor" aria-label="anchor" href="#instructors-names-and-contact-information"></a>
</h3>
<ul>
<li>Jermiah J. Joseph <a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
</li>
<li>Almas Khan <a href="mailto:almas.khan@uhn.ca" class="email">almas.khan@uhn.ca</a>
</li>
<li>Julia Nguyen <a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 class="unnumbered" id="learning-goals">Learning goals<a class="anchor" aria-label="anchor" href="#learning-goals"></a>
</h3>
<ul>
<li>Understand the data structure of a <code>PharmacoSet</code>
</li>
<li>Learn how to access features and metadata from a PharmacoSet</li>
<li>Learn how to plot batch effects using PCA and run correction method
(SVA)</li>
<li>Learn how to filter out outliers and missing values</li>
</ul>
</div>
<div class="section level3">
<h3 class="unnumbered" id="learning-objectives">Learning objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h3>
<ul>
<li>Describe the use cases for <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
in Pharmacogenomics</li>
<li>Understand the structure of the <code>CoreSet</code> and
<code>PharmacoSet</code> classes to facilitate their use in subsequent
analyses</li>
<li>Download/load a <code>PharmacoSet</code> using <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
or <a href="https://orcestra.ca" class="external-link">orcestra.ca</a>
</li>
<li>Subset and filter a <code>PharmacoSet</code> by samples and/or
treatments</li>
<li>Access the molecular features, dose-response and metadata contained
within the <code>PharmacoSet</code>
</li>
<li>Perform quality control on a <code>PharmacoSet</code> by identifying
and removing outliers and missing values</li>
<li>Perform batch correction on a <code>PharmacoSet</code> using the
<code>sva</code> package</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="section level3">
<h3 id="use-cases-for-pharmacogx">Use Cases for <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em><a class="anchor" aria-label="anchor" href="#use-cases-for-pharmacogx"></a>
</h3>
<p><em>Downloading Data from <a href="https://orcestra.ca" class="external-link">orcestra.ca</a></em></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">available</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/availablePSets.html" class="external-link">availablePSets</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/as.data.table.html" class="external-link">as.data.table</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">available</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Dataset Name" "Date Created" "PSet Name"    "version"      "type"        </span></span>
<span><span class="co">## [6] "publication"  "DOI"          "Download"</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">available</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Dataset Name"</span>, <span class="st">"PSet Name"</span>, <span class="st">"version"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     Dataset Name         PSet Name      version</span></span>
<span><span class="co">##           &lt;char&gt;            &lt;char&gt;       &lt;char&gt;</span></span>
<span><span class="co">##  1:         GDSC GDSC_2020(v2-8.2) 2020(v2-8.2)</span></span>
<span><span class="co">##  2:         FIMM         FIMM_2016         2016</span></span>
<span><span class="co">##  3:        Tavor        Tavor_2020         2020</span></span>
<span><span class="co">##  4:        NCI60        NCI60_2021         2021</span></span>
<span><span class="co">##  5:    UHNBreast    UHNBreast_2019         2019</span></span>
<span><span class="co">##  6:         GDSC GDSC_2020(v1-8.2) 2020(v1-8.2)</span></span>
<span><span class="co">##  7:        PRISM        PRISM_2020         2020</span></span>
<span><span class="co">##  8:      BeatAML      BeatAML_2018         2018</span></span>
<span><span class="co">##  9:         gCSI         gCSI_2019         2019</span></span>
<span><span class="co">## 10:       CTRPv2       CTRPv2_2015         2015</span></span>
<span><span class="co">## 11:         GRAY         GRAY_2017         2017</span></span>
<span><span class="co">## 12:         CCLE         CCLE_2015         2015</span></span>
<span><span class="co">## 13:         PDTX         PDTX_2019         2019</span></span>
<span><span class="co">## 14:          GBM          GBM_scr2         2021</span></span>
<span><span class="co">## 15:          GBM          GBM_scr3         2021</span></span>
<span><span class="co">## 16:   NCISarcoma   NCISarcoma_2015         2015</span></span></code></pre>
<p>. . .</p>
<p>The following function from <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
can be used to download any of the available PSets from <a href="https://orcestra.ca" class="external-link">orcestra.ca</a>.</p>
<p>The command to do so is:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html" class="external-link">downloadPSet</a></span><span class="op">(</span></span>
<span>  name <span class="op">=</span> <span class="st">"CCLE_2015"</span>,</span>
<span>  saveDir <span class="op">=</span> <span class="st">"../psets"</span>, <span class="co"># change this directory as you see fit</span></span>
<span>  timeout <span class="op">=</span> <span class="fl">3600</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Note: this may take a while to download as the files are stored in</span></span>
<span><span class="co"># Zenodo and are quite large</span></span></code></pre></div>
<p>. . .</p>
<p>For convenience and in the interest of time, we have created a
<code>PharmacoSet</code> to be used in this tutorial. We will be
interacting with this dataset for the remainder of the tutorial.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pset</span> <span class="op">&lt;-</span> <span class="fu">CBWWorkshop2024</span><span class="fu">::</span><span class="va"><a href="../reference/dummy_pset.html">dummy_pset</a></span></span>
<span><span class="va">pset</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">dummy_pset</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Tue Oct 15 23:31:41 2024</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">50</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: &lt;MultiAssayExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">ExperimentList class object of length 2: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [1] </span><span style="color: #BB0000;">rnaseq.tpm</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 50 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [2] </span><span style="color: #BB0000;">rnaseq.tpm.batch</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 50 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span><span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: &lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4847 52 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4847): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(52): </span><span style="color: #00BB00;">786-O:1 A-498:1 A-549:1 ... UACC-257:1 UACC-62:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="understanding-the-pharmacoset-data-structure">Understanding the <code>PharmacoSet</code> Data Structure<a class="anchor" aria-label="anchor" href="#understanding-the-pharmacoset-data-structure"></a>
</h2>
<p>The <code>PharmacoSet</code> class is a container for pharmacogenomic
data.</p>
<p>This pharmacogenomic data is typically generated from high-throughput
screening experiments where cell lines are treated with a panel of drugs
at multiple doses and the response is measured using a molecular
assay.</p>
<p>Pharmacogenomic data is located in a <code>PharmacoSet</code> within
its slots, which are accessed using the <code>@</code> operator.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">slotNames</span><span class="op">(</span><span class="va">pset</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "treatmentResponse" "annotation"        "molecularProfiles"</span></span>
<span><span class="co">## [4] "sample"            "treatment"         "datasetType"      </span></span>
<span><span class="co">## [7] "perturbation"      "curation"</span></span></code></pre>
<p>. . .</p>
<div class="figure">
<img src="CoreSet-class.png" alt="**PharmacoSet** class structure" width="742"><p class="caption">
<strong>PharmacoSet</strong> class structure
</p>
</div>
</div>
<div class="section level2">
<h2 id="metadata">Metadata<a class="anchor" aria-label="anchor" href="#metadata"></a>
</h2>
<p>Metadata for cell-lines (<code>samples</code>) and drugs
(<code>treatments</code>) are stored in the <code>sample</code> and
<code>treatment</code> slots, respectively.</p>
<p>These are data frames with rows corresponding to samples or
treatments and columns corresponding to metadata fields. The metadata
fields are typically a combination of data from the original data
source, and curated data from external sources such as <a href="https://web.expasy.org/cellosaurus/" class="external-link">Cellosaurus</a>, <a href="https://go.drugbank.com/" class="external-link">DrugBank</a>, and <a href="https://pubchem.ncbi.nlm.nih.gov/" class="external-link">PubChem</a>.</p>
<div class="section level3">
<h3 id="view-the-sample-slot">View the <code>sample</code> slot<a class="anchor" aria-label="anchor" href="#view-the-sample-slot"></a>
</h3>
<p>We can access the <code>sampleNames</code> of the
<code>PharmacoSet</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">sampleNames</a></span><span class="op">(</span><span class="va">pset</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "NCI-H23"     "HOP-62"      "SNB-75"      "HL-60(TB)"   "A-549"      </span></span>
<span><span class="co">##  [6] "SF268"       "SNB-19"      "SF295"       "OVCAR-5"     "U-251MG"    </span></span>
<span><span class="co">## [11] "MCF-7"       "MDA-MB-231"  "MDA-MB-435"  "BT-549"      "SK-MEL-5"   </span></span>
<span><span class="co">## [16] "MOLT-4"      "DU145"       "NCI-ADR-RES" "OVCAR-3"     "COLO 205"   </span></span>
<span><span class="co">## [21] "SK-OV-3"     "CCRF-CEM"    "SR"          "IGROV-1"     "SW620"      </span></span>
<span><span class="co">## [26] "Hs 578T"     "HCT 116"     "UACC-62"     "EKVX"        "HCT 15"     </span></span>
<span><span class="co">## [31] "OVCAR-8"     "UACC-257"    "SF539"       "UO-31"       "M14"        </span></span>
<span><span class="co">## [36] "KM12"        "T-47D"       "Malme-3M"    "Caki-1"      "A-498"      </span></span>
<span><span class="co">## [41] "SK-MEL-28"   "RPMI-8226"   "NCI-H460"    "NCI-H322M"   "NCI-H226"   </span></span>
<span><span class="co">## [46] "K-562"       "786-O"       "SK-MEL-2"    "HCC2998"     "NCI-H522"</span></span></code></pre>
<p>. . .</p>
<p>To get all the metadata associated with the samples, we can access
the <code>sample</code> slot:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pset</span><span class="op">@</span><span class="va">sample</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'data.frame':    50 obs. of  7 variables:</span></span>
<span><span class="co">##  $ sampleid            : chr  "NCI-H23" "HOP-62" "SNB-75" "HL-60(TB)" ...</span></span>
<span><span class="co">##  $ tissueid            : chr  "Lung" "Lung" "CNS/Brain" "Myeloid" ...</span></span>
<span><span class="co">##  $ cellosaurus_disease : chr  "Lung adenocarcinoma" "Lung adenocarcinoma" "Glioblastoma" "Acute myeloid leukemia" ...</span></span>
<span><span class="co">##  $ cellosaurus_id      : chr  "CVCL_1547" "CVCL_1285" "CVCL_1706" "CVCL_A794" ...</span></span>
<span><span class="co">##  $ NCI_ALMANAC.sampleid: chr  "NCI-H23" "HOP-62" "SNB-75" "HL-60(TB)" ...</span></span>
<span><span class="co">##  $ NCI_ALMANAC.disease : chr  "Non-Small Cell Lung Cancer" "Non-Small Cell Lung Cancer" "CNS Cancer" "Leukemia" ...</span></span>
<span><span class="co">##  $ pharmacodb_cid      : chr  "NCIH23_1085_2019" "HOP62_561_2019" "SNB75_1430_2019" "HL-60(TB)6502021_" ...</span></span></code></pre>
<p>We can see from the <code>sample</code> slot that there are 50
samples in this <code>PharmacoSet</code>.</p>
</div>
<div class="section level3">
<h3 id="view-the-treatment-slot">View the <code>treatment</code> slot<a class="anchor" aria-label="anchor" href="#view-the-treatment-slot"></a>
</h3>
<p>Similar to the <code>sample</code>, we can access the
<code>treatmentNames</code> of the <code>PharmacoSet</code>:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">treatmentNames</a></span><span class="op">(</span><span class="va">pset</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Ixabepilone" "Clofarabine" "Dasatinib"   "Vincristine" "Imiquimod"  </span></span>
<span><span class="co">## [6] "Paclitaxel"</span></span></code></pre>
<p>. . .</p>
<p>To get all the metadata associated with the treatments, we can access
the <code>treatment</code> slot:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pset</span><span class="op">@</span><span class="va">treatment</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'data.frame':    22 obs. of  3 variables:</span></span>
<span><span class="co">##  $ treatmentid: chr  "Ixabepilone" "Clofarabine" "Dasatinib" "Vincristine" ...</span></span>
<span><span class="co">##  $ cid        : int  6445540 119182 3062316 5978 57469 36314 13342 54611422 2141 176871 ...</span></span>
<span><span class="co">##  $ inchikey   : chr  "FABUFPQFXZVHFB-PVYNADRNSA-N" "WDDPHFBMKLOVOX-UHFFFAOYSA-N" "ZBNZXTGUTAYRHI-UHFFFAOYSA-N" "" ...</span></span></code></pre>
<p>There are also 22 treatments used.</p>
</div>
</div>
<div class="section level2">
<h2 id="molecular-profiles">Molecular Profiles<a class="anchor" aria-label="anchor" href="#molecular-profiles"></a>
</h2>
<p>In <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>,
molecular profiles refer to any data that is measured on the samples in
the <code>PharmacoSet</code>. This can include gene expression, copy
number, mutation, or any other type of data that can be measured on a
sample.</p>
<p>Each Molecular Data Type (<code>mDataType</code>) is stored in a
<code>SummarizedExperiment</code>. All the
<code>SummarizedExperiment</code> objects are stored in a container
called a <code>MultiAssayExperiment</code> which can be accessed through
the <code>molecularProfiles</code> slot.</p>
<div class="section level3">
<h3 id="multiassayexperiment-and-summarizedexperiment">
<code>MultiAssayExperiment</code> and
<code>SummarizedExperiment</code><a class="anchor" aria-label="anchor" href="#multiassayexperiment-and-summarizedexperiment"></a>
</h3>
<div class="figure">
<img src="MultiAssayExperiment_SummarizedExperiment_schematics.png" alt="**SummarizedExperiment** and **MultiAssayExperiment** classes." width="1053"><p class="caption">
<strong>SummarizedExperiment</strong> and
<strong>MultiAssayExperiment</strong> classes.
</p>
</div>
</div>
<div class="section level3">
<h3 id="view-the-molecularprofiles-slot">View the <code>molecularProfiles</code> slot<a class="anchor" aria-label="anchor" href="#view-the-molecularprofiles-slot"></a>
</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pset</span><span class="op">@</span><span class="va">molecularProfiles</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] rnaseq.tpm: SummarizedExperiment with 100 rows and 50 columns</span></span>
<span><span class="co">##  [2] rnaseq.tpm.batch: SummarizedExperiment with 100 rows and 50 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="treatment-response">Treatment Response<a class="anchor" aria-label="anchor" href="#treatment-response"></a>
</h2>
<p>The treatment response data was traditionally stored in a
<code>list</code>. To accelerate analysis, we have developed a new class
called the <code>TreatmentResponseExperiment</code> (TRE) which has been
specifically designed to handle high dimensional biological
stimulus-response data.</p>
<div class="section level3">
<h3 id="view-the-treatmentresponse-slot">View the <code>treatmentResponse</code> slot<a class="anchor" aria-label="anchor" href="#view-the-treatmentresponse-slot"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pset</span><span class="op">@</span><span class="va">treatmentResponse</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4847 52 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4847): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(52): </span><span style="color: #00BB00;">786-O:1 A-498:1 A-549:1 ... UACC-257:1 UACC-62:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
<p>. . .</p>
<p>To view all the methods available for the
<code>TreatmentResponseExperiment</code> class, we can use the
<code>methods</code> function:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/methods.html" class="external-link">methods</a></span><span class="op">(</span>class <span class="op">=</span> <span class="st">"TreatmentResponseExperiment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] [                   [[                  [[&lt;-               </span></span>
<span><span class="co">##  [4] $                   $&lt;-                 aggregate          </span></span>
<span><span class="co">##  [7] assay               assay&lt;-             assayCols          </span></span>
<span><span class="co">## [10] assayIndex          assayKeys           assayNames         </span></span>
<span><span class="co">## [13] assays              assays&lt;-            buildComboProfiles </span></span>
<span><span class="co">## [16] coerce              colData             colData&lt;-          </span></span>
<span><span class="co">## [19] colIDs              colMeta             colnames           </span></span>
<span><span class="co">## [22] computeZIPdelta     dim                 dimnames           </span></span>
<span><span class="co">## [25] endoaggregate       getIntern           idCols             </span></span>
<span><span class="co">## [28] mergeAssays         metadata            metadata&lt;-         </span></span>
<span><span class="co">## [31] names               reindex             rowData            </span></span>
<span><span class="co">## [34] rowData&lt;-           rowIDs              rowMeta            </span></span>
<span><span class="co">## [37] rownames            show                subset             </span></span>
<span><span class="co">## [40] treatmentResponse&lt;- updateObject       </span></span>
<span><span class="co">## see '?methods' for accessing help and source code</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="subsetting-a-pharmacoset">Subsetting a <code>PharmacoSet</code><a class="anchor" aria-label="anchor" href="#subsetting-a-pharmacoset"></a>
</h2>
<div class="section level3">
<h3 id="subset-by-samplenames-andor-treatmentnames">Subset by <code>sampleNames</code> and/or
<code>treatmentNames</code><a class="anchor" aria-label="anchor" href="#subset-by-samplenames-andor-treatmentnames"></a>
</h3>
<p>When subsetting a <code>PharmacoSet</code> object by
<code>sample</code> names, both the <code>MultiAssayExperiment</code>
and <code>TreatmentResponseExperiment</code> objects will be subsetted
so all their internal <code>Experiments</code> only contain data for the
samples of interest.</p>
<p>However, when subsetting by <code>treatment</code> names, only the
<code>TRE</code> object will be subsetted.</p>
<p>. . .</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-utils.html" class="external-link">subsetBySample</a></span><span class="op">(</span></span>
<span>  <span class="va">pset</span>,</span>
<span>  sample <span class="op">=</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">sampleNames</a></span><span class="op">(</span><span class="va">pset</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">dummy_pset</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Tue Oct 15 23:31:41 2024</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">5</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: &lt;MultiAssayExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">ExperimentList class object of length 2: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [1] </span><span style="color: #BB0000;">rnaseq.tpm</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 5 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [2] </span><span style="color: #BB0000;">rnaseq.tpm.batch</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 5 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span><span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: &lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4440 4 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4440): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(4): </span><span style="color: #00BB00;">A-549:1 HOP-62:1 NCI-H23:1 SNB-75:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-utils.html" class="external-link">subsetByTreatment</a></span><span class="op">(</span></span>
<span>  <span class="va">pset</span>,</span>
<span>  treatment <span class="op">=</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">treatmentNames</a></span><span class="op">(</span><span class="va">pset</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">dummy_pset</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Tue Oct 15 23:31:41 2024</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">50</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: &lt;MultiAssayExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">ExperimentList class object of length 2: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [1] </span><span style="color: #BB0000;">rnaseq.tpm</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 50 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [2] </span><span style="color: #BB0000;">rnaseq.tpm.batch</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 50 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span><span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: &lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  883 51 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(883): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 7:Clofarabine:1 10:Clofarabine:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(51): </span><span style="color: #00BB00;">786-O:1 A-498:1 A-549:1 ... UACC-257:1 UACC-62:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="subsetting-using-sample-and-treatment-metadata">Subsetting using <code>sample</code> and <code>treatment</code>
metadata<a class="anchor" aria-label="anchor" href="#subsetting-using-sample-and-treatment-metadata"></a>
</h3>
<p>The utility of the <code>metadata</code> slots in the
<code>PharmacoSet</code> class</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">pset</span><span class="op">@</span><span class="va">sample</span><span class="op">$</span><span class="va">tissueid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "Lung"                 "CNS/Brain"            "Myeloid"             </span></span>
<span><span class="co">##  [4] "Ovary/Fallopian Tube" "Breast"               "Skin"                </span></span>
<span><span class="co">##  [7] "Lymphoid"             "Prostate"             "Bowel"               </span></span>
<span><span class="co">## [10] "Kidney"               "Pleura"</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tissues_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Kidney"</span><span class="op">)</span></span>
<span></span>
<span><span class="op">(</span><span class="va">samples_of_interest</span> <span class="op">&lt;-</span> <span class="va">pset</span><span class="op">@</span><span class="va">sample</span><span class="op">[</span></span>
<span>  <span class="va">pset</span><span class="op">@</span><span class="va">sample</span><span class="op">$</span><span class="va">tissueid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">tissues_of_interest</span>,</span>
<span>  <span class="st">"sampleid"</span></span>
<span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "UO-31"  "Caki-1" "A-498"  "786-O"</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-utils.html" class="external-link">subsetBySample</a></span><span class="op">(</span></span>
<span>  <span class="va">pset</span>,</span>
<span>  <span class="va">samples_of_interest</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">dummy_pset</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Tue Oct 15 23:31:41 2024</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">4</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: &lt;MultiAssayExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">ExperimentList class object of length 2: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [1] </span><span style="color: #BB0000;">rnaseq.tpm</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 4 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [2] </span><span style="color: #BB0000;">rnaseq.tpm.batch</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 4 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span><span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: &lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4285 4 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4285): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(4): </span><span style="color: #00BB00;">786-O:1 A-498:1 Caki-1:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
<p>A concise way to subset by <code>sample</code> in one step:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-utils.html" class="external-link">subsetBySample</a></span><span class="op">(</span></span>
<span>  <span class="va">pset</span>,</span>
<span>  sample <span class="op">=</span> <span class="va">pset</span><span class="op">@</span><span class="va">sample</span><span class="op">[</span></span>
<span>    <span class="va">pset</span><span class="op">@</span><span class="va">sample</span><span class="op">$</span><span class="va">tissueid</span> <span class="op">==</span> <span class="st">"Kidney"</span>,</span>
<span>    <span class="st">"sampleid"</span></span>
<span>  <span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">dummy_pset</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Tue Oct 15 23:31:41 2024</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">4</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: &lt;MultiAssayExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">ExperimentList class object of length 2: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [1] </span><span style="color: #BB0000;">rnaseq.tpm</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 4 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span>   <span style="color: #BBBB00; font-weight: bold;"> [2] </span><span style="color: #BB0000;">rnaseq.tpm.batch</span> <span style="color: #00BB00;">: SummarizedExperiment with 100 rows and 4 columns </span></span></span>
<span><span class="co"><span style="color: #00BB00;">## </span><span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: &lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4285 4 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4285): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(4): </span><span style="color: #00BB00;">786-O:1 A-498:1 Caki-1:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="computing-over-a-treatmentresponseexperiment">Computing over a <code>TreatmentResponseExperiment</code><a class="anchor" aria-label="anchor" href="#computing-over-a-treatmentresponseexperiment"></a>
</h2>
<p>The <code>TreatmentResponseExperiment</code> class has been designed
to facilitate more than just storing data efficiently. Once the data is
curated and stored safely inside our object, we are ready to start
answering questions.</p>
<p>First, let’s extract the <code>TRE</code> to work with:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre</span> <span class="op">&lt;-</span> <span class="va">pset</span><span class="op">@</span><span class="va">treatmentResponse</span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="va">tre</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4847 52 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(1): </span><span style="color: #BB0000;">sensitivity</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4847): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(52): </span><span style="color: #00BB00;">786-O:1 A-498:1 A-549:1 ... UACC-257:1 UACC-62:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
<div class="section level3">
<h3 id="aggregating-over-the-treatmentresponseexperiment">Aggregating over the <code>TreatmentResponseExperiment</code><a class="anchor" aria-label="anchor" href="#aggregating-over-the-treatmentresponseexperiment"></a>
</h3>
<p>The <code>aggregate</code> function is a powerful tool that allows us
to summarize the data in a <code>TreatmentResponseExperiment</code>
object.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span></span>
<span>    assay <span class="op">=</span> <span class="st">"sensitivity"</span>,</span>
<span>    N <span class="op">=</span> <span class="va">.N</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"treatmentdose"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       treatmentid treatmentdose sampleid bio_rep     N</span></span>
<span><span class="co">##            &lt;char&gt;         &lt;num&gt;   &lt;char&gt;   &lt;int&gt; &lt;int&gt;</span></span>
<span><span class="co">##    1: Ixabepilone         1e-07    786-O       1     1</span></span>
<span><span class="co">##    2: Ixabepilone         1e-07    A-498       1     1</span></span>
<span><span class="co">##    3: Ixabepilone         1e-07    A-549       1     1</span></span>
<span><span class="co">##    4: Ixabepilone         1e-07   BT-549       1     1</span></span>
<span><span class="co">##    5: Ixabepilone         1e-07 COLO 205       1     1</span></span>
<span><span class="co">##   ---                                                 </span></span>
<span><span class="co">## 9036:  Ifosfamide         4e+03    T-47D       1     1</span></span>
<span><span class="co">## 9037:  Ifosfamide         4e+03  U-251MG       1     1</span></span>
<span><span class="co">## 9038:  Ifosfamide         4e+03 UACC-257       1     1</span></span>
<span><span class="co">## 9039:  Ifosfamide         4e+03  UACC-62       1     1</span></span>
<span><span class="co">## 9040:  Ifosfamide         4e+03    UO-31       1     1</span></span></code></pre>
<p>You can also use the <code>subset</code> function to subset the
<code>TreatmentResponseExperiment</code> object before performing
<code>aggregate</code>.</p>
<p>Additionally, you can even <code>subset</code> over the result of an
<code>aggregate</code> operation.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">treatmentid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Doxorubicin"</span>, <span class="st">"Ixabepilone"</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span></span>
<span>    assay <span class="op">=</span> <span class="st">"sensitivity"</span>,</span>
<span>    N <span class="op">=</span> <span class="va">.N</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"treatmentdose"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">N</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">sampleid</span> <span class="op">==</span> <span class="st">"A-498"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     treatmentid treatmentdose sampleid bio_rep     N</span></span>
<span><span class="co">##          &lt;char&gt;         &lt;num&gt;   &lt;char&gt;   &lt;int&gt; &lt;int&gt;</span></span>
<span><span class="co">##  1: Ixabepilone         3e-05    A-498       1     2</span></span>
<span><span class="co">##  2: Doxorubicin         5e-05    A-498       1   102</span></span>
<span><span class="co">##  3: Ixabepilone         3e-04    A-498       1     2</span></span>
<span><span class="co">##  4: Doxorubicin         5e-04    A-498       1   102</span></span>
<span><span class="co">##  5: Ixabepilone         3e-03    A-498       1    10</span></span>
<span><span class="co">##  6: Doxorubicin         5e-03    A-498       1   122</span></span>
<span><span class="co">##  7: Ixabepilone         3e-02    A-498       1     8</span></span>
<span><span class="co">##  8: Doxorubicin         5e-02    A-498       1   122</span></span>
<span><span class="co">##  9: Ixabepilone         3e-01    A-498       1     8</span></span>
<span><span class="co">## 10: Doxorubicin         5e-01    A-498       1   122</span></span>
<span><span class="co">## 11: Doxorubicin         5e+00    A-498       1   102</span></span></code></pre>
<div class="section level4">
<h4 id="plotting-the-number-of-replicates">Plotting the number of replicates<a class="anchor" aria-label="anchor" href="#plotting-the-number-of-replicates"></a>
</h4>
<p>The last two examples have shown that there are technical replicates
in the data. We can visualize the number of replicates using a
histogram.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span></span>
<span>    assay <span class="op">=</span> <span class="st">"sensitivity"</span>,</span>
<span>    N <span class="op">=</span> <span class="va">.N</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"treatmentdose"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span></span>
<span>    expr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span></span>
<span>      <span class="va">N</span>,</span>
<span>      main <span class="op">=</span> <span class="st">"Histogram of Number of Replicates"</span>,</span>
<span>      xlab <span class="op">=</span> <span class="st">"Number of Replicates"</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="Module2_files/figure-html/tre_aggregate-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="summarize-over-technical-replicates">Summarize over technical replicates<a class="anchor" aria-label="anchor" href="#summarize-over-technical-replicates"></a>
</h4>
<p>Depending on the experiment design, you will need to choose how you
want to proceed with the data.</p>
<p>One common approach is to summarize the technical replicates into a
single value, this can be done by taking the mean, median, or any other
summary statistic.</p>
</div>
</div>
<div class="section level3">
<h3 id="endoaggregate">
<code>Endoaggregate</code><a class="anchor" aria-label="anchor" href="#endoaggregate"></a>
</h3>
<p>In the interest of computation time and resources, we want to
summarize the technical replicates into a single value and store that
result <strong>inside</strong> the
<code>TreatmentResponseExperiment</code>.</p>
<p>For this, we have developed the <code>endoaggregate</code>
method.</p>
<p>This method will perform similar to the <code>aggregate</code>
method, but will return a <code>TreatmentResponseExperiment</code>
object with the summarized data as another assay.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_tre</span> <span class="op">&lt;-</span> <span class="va">tre</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">endoaggregate</span><span class="op">(</span></span>
<span>    N <span class="op">=</span> <span class="va">.N</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"sensitivity"</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"tech_rep_counts"</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"treatmentdose"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu">show</span><span class="op">(</span><span class="va">new_tre</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4847 52 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(2): </span><span style="color: #BB0000;">sensitivity tech_rep_counts</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4847): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(52): </span><span style="color: #00BB00;">786-O:1 A-498:1 A-549:1 ... UACC-257:1 UACC-62:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">new_tre</span><span class="op">$</span><span class="va">tech_rep_counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">N</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    treatmentdose treatmentid    sampleid bio_rep     N</span></span>
<span><span class="co">##            &lt;num&gt;      &lt;char&gt;      &lt;char&gt;   &lt;int&gt; &lt;int&gt;</span></span>
<span><span class="co">## 1:         0.005 Doxorubicin     HCT 116       1   153</span></span>
<span><span class="co">## 2:         0.050 Doxorubicin     HCT 116       1   153</span></span>
<span><span class="co">## 3:         0.500 Doxorubicin     HCT 116       1   153</span></span>
<span><span class="co">## 4:         0.005 Doxorubicin NCI-ADR-RES       1   132</span></span>
<span><span class="co">## 5:         0.050 Doxorubicin NCI-ADR-RES       1   132</span></span>
<span><span class="co">## 6:         0.500 Doxorubicin NCI-ADR-RES       1   132</span></span></code></pre>
<p>Let’s use this method to summarize the technical replicates by taking
the mean of the <code>viability</code> values for each unique
combination of <code>treatmentid</code>, <code>treatmentdose</code>,
<code>sampleid</code>, and <code>bio_rep</code>.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre_avged</span> <span class="op">&lt;-</span> <span class="va">tre</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">endoaggregate</span><span class="op">(</span></span>
<span>    viability <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">viability</span><span class="op">)</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"sensitivity"</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"mean_sensitivity"</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"treatmentdose"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">tre_avged</span><span class="op">$</span><span class="va">mean_sensitivity</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Classes 'data.table' and 'data.frame':   9040 obs. of  5 variables:</span></span>
<span><span class="co">##  $ treatmentdose: num  1e-07 1e-07 1e-07 1e-07 1e-07 1e-07 1e-07 1e-07 1e-07 1e-07 ...</span></span>
<span><span class="co">##  $ treatmentid  : chr  "Ixabepilone" "Ixabepilone" "Ixabepilone" "Ixabepilone" ...</span></span>
<span><span class="co">##  $ sampleid     : chr  "786-O" "A-498" "A-549" "BT-549" ...</span></span>
<span><span class="co">##  $ bio_rep      : int  1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span><span class="co">##  $ viability    : num  98 99.7 99.3 96 103.7 ...</span></span>
<span><span class="co">##  - attr(*, "sorted")= chr [1:4] "treatmentdose" "treatmentid" "sampleid" "bio_rep"</span></span>
<span><span class="co">##  - attr(*, ".internal.selfref")=&lt;externalptr&gt;</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="computing-profiles">Computing profiles<a class="anchor" aria-label="anchor" href="#computing-profiles"></a>
</h3>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre_profiled</span> <span class="op">&lt;-</span> <span class="va">tre_avged</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">endoaggregate</span><span class="op">(</span></span>
<span>    <span class="op">{</span></span>
<span>      <span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/logLogisticRegression.html" class="external-link">logLogisticRegression</a></span><span class="op">(</span><span class="va">treatmentdose</span>, <span class="va">viability</span><span class="op">)</span></span>
<span>      <span class="va">IC50</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeICn.html" class="external-link">computeIC50</a></span><span class="op">(</span><span class="va">treatmentdose</span>, Hill_fit <span class="op">=</span> <span class="va">fit</span><span class="op">)</span></span>
<span>      <span class="va">AAC</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeAUC.html" class="external-link">computeAUC</a></span><span class="op">(</span><span class="va">treatmentdose</span>, Hill_fit <span class="op">=</span> <span class="va">fit</span>, area.type <span class="op">=</span> <span class="st">"Fitted"</span><span class="op">)</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        HS <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"HS"</span><span class="op">]</span><span class="op">]</span>, E_inf <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"E_inf"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="fl">100</span>, EC50 <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"EC50"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>        Rsq <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        AAC <span class="op">=</span> <span class="va">AAC</span>,</span>
<span>        IC50 <span class="op">=</span> <span class="va">IC50</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">}</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"mean_sensitivity"</span>,</span>
<span>    subset <span class="op">=</span> <span class="va">treatmentid</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu">rowData</span><span class="op">(</span><span class="va">tre_avged</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">treatmentid</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span>,</span>
<span>    enlist <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"profiles"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>If we didn’t subset the <code>TRE</code> this computation will have
taken a while since it is a computationally intensive operation and run
on a single thread.</p>
<p>Under the hood, the <code>CoreGx</code> package performs the
aggregation. To speed up computation, it provides an
<code>nthread</code> parameter to parallelize the computation.</p>
<p>To determine a safe, and optimal number of threads to use, we can use
the <code><a href="https://rdrr.io/pkg/CoreGx/man/optimizeCoreGx.html" class="external-link">CoreGx::optimizeCoreGx</a></code> function which will tell us a
safe number of threads.</p>
<p>NOTE: The following two code chunks will not be evaluated in this
vignette as it may take some time.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">CoreGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/optimizeCoreGx.html" class="external-link">optimizeCoreGx</a></span><span class="op">(</span><span class="va">tre_avged</span><span class="op">)</span></span></code></pre></div>
<p>Depending on your device, this number may vary. Let’s use a
conservative number of threads for this example.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">THREADS</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># github runners provide 2c</span></span>
<span></span>
<span><span class="va">tre_profiled</span> <span class="op">&lt;-</span> <span class="va">tre_avged</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">endoaggregate</span><span class="op">(</span></span>
<span>    <span class="op">{</span></span>
<span>      <span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/logLogisticRegression.html" class="external-link">logLogisticRegression</a></span><span class="op">(</span><span class="va">treatmentdose</span>, <span class="va">viability</span><span class="op">)</span></span>
<span>      <span class="va">IC50</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeICn.html" class="external-link">computeIC50</a></span><span class="op">(</span><span class="va">treatmentdose</span>, Hill_fit <span class="op">=</span> <span class="va">fit</span><span class="op">)</span></span>
<span>      <span class="va">AAC</span> <span class="op">&lt;-</span> <span class="fu">PharmacoGx</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeAUC.html" class="external-link">computeAUC</a></span><span class="op">(</span><span class="va">treatmentdose</span>, Hill_fit <span class="op">=</span> <span class="va">fit</span>, area.type <span class="op">=</span> <span class="st">"Fitted"</span><span class="op">)</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        HS <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"HS"</span><span class="op">]</span><span class="op">]</span>, E_inf <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"E_inf"</span><span class="op">]</span><span class="op">]</span> <span class="op">/</span> <span class="fl">100</span>, EC50 <span class="op">=</span> <span class="va">fit</span><span class="op">[[</span><span class="st">"EC50"</span><span class="op">]</span><span class="op">]</span>,</span>
<span>        Rsq <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        AAC <span class="op">=</span> <span class="va">AAC</span>,</span>
<span>        IC50 <span class="op">=</span> <span class="va">IC50</span>,</span>
<span>        mindose <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">treatmentdose</span><span class="op">)</span>,</span>
<span>        maxdose <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">treatmentdose</span><span class="op">)</span>,</span>
<span>        numdose <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">treatmentdose</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        minviability <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">viability</span><span class="op">)</span>,</span>
<span>        maxviability <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">viability</span><span class="op">)</span></span>
<span>      <span class="op">)</span></span>
<span>    <span class="op">}</span>,</span>
<span>    assay <span class="op">=</span> <span class="st">"mean_sensitivity"</span>,</span>
<span>    by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"treatmentid"</span>, <span class="st">"sampleid"</span>, <span class="st">"bio_rep"</span><span class="op">)</span>,</span>
<span>    enlist <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    target <span class="op">=</span> <span class="st">"profiles"</span>,</span>
<span>    nthread <span class="op">=</span> <span class="va">THREADS</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Let’s look at the profiles we have computed:</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">show</span><span class="op">(</span><span class="va">tre_profiled</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;TreatmentResponseExperiment&gt; </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>   <span style="color: #BBBB00; font-weight: bold;">dim:  4847 52 </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold;">## </span>   <span style="color: #BBBB00; font-weight: bold;">assays(3): </span><span style="color: #BB0000;">sensitivity mean_sensitivity profiles</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rownames(4847): </span><span style="color: #00BB00;">1e-07:Ixabepilone:1 1e-06:Ixabepilone:1 ... 1000:Nelarabine:1 4000:Ifosfamide:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">rowData(4): </span><span style="color: #00BB00;">treatmentdose treatmentid tech_rep CONC1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colnames(52): </span><span style="color: #00BB00;">786-O:1 A-498:1 A-549:1 ... UACC-257:1 UACC-62:1 UO-31:1</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">colData(3): </span><span style="color: #00BB00;">sampleid bio_rep PANEL</span> </span></span>
<span><span class="co">##    <span style="color: #BBBB00; font-weight: bold;">metadata(0): </span><span style="color: #00BB00;">none</span></span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tre_profiled</span><span class="op">$</span><span class="va">profiles</span></span></code></pre></div>
<pre><code><span><span class="co">## Key: &lt;treatmentid, sampleid, bio_rep&gt;</span></span>
<span><span class="co">##      treatmentid sampleid bio_rep    HS      E_inf  EC50       Rsq       AAC</span></span>
<span><span class="co">##           &lt;char&gt;   &lt;char&gt;   &lt;int&gt; &lt;num&gt;      &lt;num&gt; &lt;num&gt;     &lt;num&gt;     &lt;num&gt;</span></span>
<span><span class="co">##   1: Ixabepilone    786-O       1     1 0.22057530  0.03 0.9794689 12.531587</span></span>
<span><span class="co">##   2: Ixabepilone    A-498       1     1 0.31290066  0.03 0.9789242 11.047180</span></span>
<span><span class="co">##   3: Ixabepilone    A-549       1     1 0.14355321  0.03 0.9457645 13.769947</span></span>
<span><span class="co">##   4: Ixabepilone   BT-549       1     1 0.20976649  0.03 0.9608561 12.705370</span></span>
<span><span class="co">##   5: Ixabepilone CCRF-CEM       1     1 0.02700157  0.03 0.8583904 25.321277</span></span>
<span><span class="co">##  ---                                                                        </span></span>
<span><span class="co">## 147: Vinblastine    T-47D       1     1 0.54684798  0.05 0.3701465  5.404228</span></span>
<span><span class="co">## 148: Vinblastine  U-251MG       1     1 0.23576667  0.01 0.5443226 19.888382</span></span>
<span><span class="co">## 149: Vinblastine UACC-257       1     1 0.50895401  0.05 0.3843545  5.856146</span></span>
<span><span class="co">## 150: Vinblastine  UACC-62       1     1 0.31347623  0.01 0.5809319 17.866071</span></span>
<span><span class="co">## 151: Vinblastine    UO-31       1     1 0.68541114  0.10 0.6336420  2.367176</span></span>
<span><span class="co">##            IC50</span></span>
<span><span class="co">##           &lt;num&gt;</span></span>
<span><span class="co">##   1: 0.05368172</span></span>
<span><span class="co">##   2: 0.08017131</span></span>
<span><span class="co">##   3: 0.04208202</span></span>
<span><span class="co">##   4: 0.05168252</span></span>
<span><span class="co">##   5: 0.03171258</span></span>
<span><span class="co">##  ---           </span></span>
<span><span class="co">## 147:         NA</span></span>
<span><span class="co">## 148: 0.01892267</span></span>
<span><span class="co">## 149:         NA</span></span>
<span><span class="co">## 150: 0.02680624</span></span>
<span><span class="co">## 151:         NA</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="quality-control-of-pharmacogenomic-datasets">Quality Control of Pharmacogenomic Datasets<a class="anchor" aria-label="anchor" href="#quality-control-of-pharmacogenomic-datasets"></a>
</h2>
<p>For the remainder of the lab, we briefly go over some common
scenarios that require quality control. We will load in a new molecular
profile.</p>
<div class="section level3">
<h3 id="checking-for-missing-values">Checking for missing values<a class="anchor" aria-label="anchor" href="#checking-for-missing-values"></a>
</h3>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get molecular profile</span></span>
<span><span class="va">tpm_matrix</span> <span class="op">&lt;-</span> <span class="va">pset</span><span class="op">@</span><span class="va">molecularProfiles</span><span class="op">[[</span><span class="st">"rnaseq.tpm.batch"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">assay</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tpm_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            NCI-H23    HOP-62     SNB-75 HL-60(TB)     A-549</span></span>
<span><span class="co">## GENE001 90.5370190  75.59677 104.525210  42.18450  9.391661</span></span>
<span><span class="co">## GENE002 75.5065358  22.05942         NA        NA 82.203007</span></span>
<span><span class="co">## GENE003 85.2218882 103.63729   8.646855  86.64833 42.742828</span></span>
<span><span class="co">## GENE004  0.0749728  89.58816  19.705347  45.51081 75.588726</span></span>
<span><span class="co">## GENE005 57.0379889  80.19416  92.440089  53.37649 66.238551</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get metadata</span></span>
<span><span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">pset</span><span class="op">@</span><span class="va">molecularProfiles</span><span class="op">[[</span><span class="st">"rnaseq.tpm.batch"</span><span class="op">]</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu">colData</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">meta</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 2 columns</span></span>
<span><span class="co">##              sampleid   batchid</span></span>
<span><span class="co">##           &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">## NCI-H23       NCI-H23         1</span></span>
<span><span class="co">## HOP-62         HOP-62         1</span></span>
<span><span class="co">## SNB-75         SNB-75         1</span></span>
<span><span class="co">## HL-60(TB)   HL-60(TB)         2</span></span>
<span><span class="co">## A-549           A-549         2</span></span>
<span><span class="co">## SF268           SF268         2</span></span></code></pre>
<p>For this exercise, we purposely put in a gene with many missing
values. Let’s look for this gene.</p>
<p>The code below will quantify the number of <code>NA</code> values in
each row (gene).</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">tpm_matrix</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GENE001 GENE002 GENE003 GENE004 GENE005 GENE006 </span></span>
<span><span class="co">##       0      25       0       0       0       0</span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print only genes with NA values</span></span>
<span><span class="va">sum_NA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">tpm_matrix</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sum_NA</span><span class="op">[</span><span class="va">sum_NA</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GENE002 </span></span>
<span><span class="co">##      25</span></span></code></pre>
<p>We see that <code>GENE002</code> has 25 <code>NA</code> values. It is
important to consider the proportion of <code>NA</code> values relative
to the number of observations. We can quickly calculate the
proportion:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">num_NA</span> <span class="op">&lt;-</span> <span class="va">sum_NA</span><span class="op">[</span><span class="va">sum_NA</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">num_observations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">tpm_matrix</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get proportion</span></span>
<span><span class="va">num_NA</span> <span class="op">/</span> <span class="va">num_observations</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 0.5</span></span></code></pre>
<p>Now we know that 50% of samples have an <code>NA</code> value rather
than an expression level for <code>GENE002</code>. This can be
indicative of some technical issue. Let’s remove this gene from
downstream analysis.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tpm_matrix</span> <span class="op">&lt;-</span> <span class="va">tpm_matrix</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tpm_matrix</span><span class="op">)</span> <span class="op">==</span> <span class="st">"GENE002"</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">tpm_matrix</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] FALSE</span></span></code></pre>
<p>In the last line of code, we check if there were any remaining
<code>NA</code> values. Since there are none, we can move on to
visualizing our data.</p>
</div>
<div class="section level3">
<h3 id="principal-component-analysis-for-outlier-removal">Principal component analysis for outlier removal<a class="anchor" aria-label="anchor" href="#principal-component-analysis-for-outlier-removal"></a>
</h3>
<p>Let’s quickly take a look at our expression matrix again.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tpm_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##            NCI-H23    HOP-62     SNB-75 HL-60(TB)     A-549</span></span>
<span><span class="co">## GENE001 90.5370190  75.59677 104.525210  42.18450  9.391661</span></span>
<span><span class="co">## GENE003 85.2218882 103.63729   8.646855  86.64833 42.742828</span></span>
<span><span class="co">## GENE004  0.0749728  89.58816  19.705347  45.51081 75.588726</span></span>
<span><span class="co">## GENE005 57.0379889  80.19416  92.440089  53.37649 66.238551</span></span>
<span><span class="co">## GENE006 26.4142662  74.16214  88.222117  96.38433 44.452740</span></span></code></pre>
<p>The expression matrix is currently formatted such that the features
(genes) are the rows and the samples (cells) are the columns.</p>
<p>Since we want to do PCA to visualize the <em>samples</em>, we need to
transpose the matrix so that the attributes of the samples (genes) are
the columns instead.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">t_tpm_matrix</span> <span class="op">&lt;-</span> <span class="va">tpm_matrix</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">t_tpm_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##              GENE001    GENE003    GENE004  GENE005  GENE006</span></span>
<span><span class="co">## NCI-H23    90.537019  85.221888  0.0749728 57.03799 26.41427</span></span>
<span><span class="co">## HOP-62     75.596766 103.637293 89.5881605 80.19416 74.16214</span></span>
<span><span class="co">## SNB-75    104.525210   8.646855 19.7053470 92.44009 88.22212</span></span>
<span><span class="co">## HL-60(TB)  42.184495  86.648331 45.5108051 53.37649 96.38433</span></span>
<span><span class="co">## A-549       9.391661  42.742828 75.5887260 66.23855 44.45274</span></span></code></pre>
<p>Once the expression matrix is in the correct format, we can perform
PCA using the <code>prcomp</code> function.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">org_pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="va">t_tpm_matrix</span><span class="op">)</span><span class="op">$</span><span class="va">x</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">org_pca</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 50 50</span></span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">org_pca</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                  PC1       PC2       PC3        PC4       PC5</span></span>
<span><span class="co">## NCI-H23   -44.039730 -45.92050  63.97547  -9.390331  42.39060</span></span>
<span><span class="co">## HOP-62     -9.862044  10.97767 -90.24025  40.301276 -27.50330</span></span>
<span><span class="co">## SNB-75     16.981835 -43.81072 -16.12549  -1.423588 -42.08150</span></span>
<span><span class="co">## HL-60(TB)  83.651364  10.58870  59.28436 -24.431547   6.36504</span></span>
<span><span class="co">## A-549       8.510913  84.27694  59.59322  -9.085760  29.08710</span></span></code></pre>
<p>To visualize the data, we plot the first two principal components
(PCs).</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># add batch labels</span></span>
<span><span class="va">org_pca</span><span class="op">$</span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">$</span><span class="va">batchid</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">org_pca</span><span class="op">)</span>, <span class="va">meta</span><span class="op">$</span><span class="va">sampleid</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">org_pca</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, color <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#624763"</span>, <span class="st">"#B1D3A3"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Before Batch Correction"</span><span class="op">)</span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p><img src="Module2_files/figure-html/plot_PCs_batch-1.png" width="700"></p>
<p>There is a very obvious outlier in our data. We can quantitatively
identify this outlier by identifying samples that lie beyond a threshold
of <code>1.5 * IQR</code> of the first and fourth quartiles.</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># extract PC scores for given PC</span></span>
<span><span class="va">scores</span> <span class="op">&lt;-</span> <span class="va">org_pca</span><span class="op">[[</span><span class="st">"PC1"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># compute IQR of given PC</span></span>
<span><span class="va">iqr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/IQR.html" class="external-link">IQR</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># compute upper and lower thresholds</span></span>
<span><span class="va">upp_thres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">scores</span>, <span class="fl">0.75</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1.5</span> <span class="op">*</span> <span class="va">iqr</span></span>
<span><span class="va">low_thres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">scores</span>, <span class="fl">0.25</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1.5</span> <span class="op">*</span> <span class="va">iqr</span></span>
<span></span>
<span><span class="co"># print samples with values beyond threshold</span></span>
<span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">org_pca</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">scores</span> <span class="op">&lt;</span> <span class="va">low_thres</span> <span class="op">|</span> <span class="va">scores</span> <span class="op">&gt;</span> <span class="va">upp_thres</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Outlier for PC1:"</span>, <span class="va">outlier</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Outlier for PC1: T-47D"</span></span></code></pre>
<p>Let’s remove <code>T-47D</code> and re-plot our data.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># remove outlier</span></span>
<span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="st">"T-47D"</span></span>
<span><span class="va">tpm_matrix</span><span class="op">[[</span><span class="va">outlier</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span></span>
<span><span class="co"># redo PCA and plotting</span></span>
<span><span class="va">org_pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">tpm_matrix</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">x</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">org_pca</span><span class="op">$</span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">$</span><span class="va">batchid</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">org_pca</span><span class="op">)</span>, <span class="va">meta</span><span class="op">$</span><span class="va">sampleid</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">org_pca</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, color <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#624763"</span>, <span class="st">"#B1D3A3"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"Before Batch Correction"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p><img src="Module2_files/figure-html/rm_outliers-1.png" width="700"></p>
<p>Now that the outlier is removed, we have a better visualization of
the data. It is very evident that the data points (samples) tend to
cluster by batch. This is indicative of a potential batch effect.</p>
</div>
<div class="section level3">
<h3 id="adjusting-for-known-batches">Adjusting for known batches<a class="anchor" aria-label="anchor" href="#adjusting-for-known-batches"></a>
</h3>
<p>When we have a known batch, we can adjust for it directly. One method
is to use the <code>ComBat()</code> function from the <code>sva</code>
package.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># adjust for known batch</span></span>
<span><span class="va">adj_tpm_matrix</span> <span class="op">&lt;-</span> <span class="fu">ComBat</span><span class="op">(</span></span>
<span>  dat <span class="op">=</span> <span class="va">tpm_matrix</span>,</span>
<span>  batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">batchid</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tpm_matrix</span><span class="op">)</span>, <span class="va">meta</span><span class="op">$</span><span class="va">sampleid</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  par.prior <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  prior.plots <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Found2batches</span></span></code></pre>
<pre><code><span><span class="co">## Adjusting for0covariate(s) or covariate level(s)</span></span></code></pre>
<pre><code><span><span class="co">## Standardizing Data across genes</span></span></code></pre>
<pre><code><span><span class="co">## Fitting L/S model and finding priors</span></span></code></pre>
<pre><code><span><span class="co">## Finding parametric adjustments</span></span></code></pre>
<pre><code><span><span class="co">## Adjusting the Data</span></span></code></pre>
<div class="sourceCode" id="cb87"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adj_tpm_matrix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##           NCI-H23   HOP-62    SNB-75 HL-60(TB)     A-549</span></span>
<span><span class="co">## GENE001 81.991219 68.65465 94.477925  46.14868  9.844954</span></span>
<span><span class="co">## GENE003 78.009018 94.55152  9.221893  94.18421 46.353576</span></span>
<span><span class="co">## GENE004  2.112512 81.72158 19.570898  48.84136 82.434411</span></span>
<span><span class="co">## GENE005 53.126927 73.84929 84.808130  57.36965 71.508314</span></span>
<span><span class="co">## GENE006 25.252026 67.80845 80.339726 106.46140 48.702080</span></span></code></pre>
<p>We can now perform PCA and visualize on our adjusted TPM counts
matrix. We will compare the results to the original PCA
visualization.</p>
<p>Don’t forget to transpose the adjusted counts matrix!</p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># perform PCA on adjusted matrix</span></span>
<span></span>
<span><span class="va">adj_pca</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">prcomp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">adj_tpm_matrix</span><span class="op">)</span><span class="op">)</span><span class="op">$</span><span class="va">x</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">adj_pca</span><span class="op">$</span><span class="va">Batch</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">$</span><span class="va">batchid</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">adj_pca</span><span class="op">)</span>, <span class="va">meta</span><span class="op">$</span><span class="va">sampleid</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># create scatter plot</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">adj_pca</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span>, color <span class="op">=</span> <span class="va">Batch</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_color_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#624763"</span>, <span class="st">"#B1D3A3"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">"After Batch Correction"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot both plots side by side for comparison</span></span>
<span><span class="fu">ggarrange</span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="Module2_files/figure-html/pca_adj_batch-1.png" width="700"></p>
<p>For downstream pharmacogenomic analysis (or any other bioinformatic
analysis), the batch-corrected expression matrix should be used.</p>
<p>For downstream pharmacogenomic analysis (or any other bioinformatic
analysis), the batch-corrected expression matrix should be used.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jermiah Joseph, Julia Nguyen, Nikta Feizi, Canadian Bioinformatics Workshop,Bioinformatics.ca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
