<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Module 3: Pharmacogenomics for biomarker discovery – Basic analysis • CBWWorkshop2024</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Module 3: Pharmacogenomics for biomarker discovery – Basic analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CBWWorkshop2024</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9027</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Introduction</h6></li>
    <li><a class="dropdown-item" href="../articles/PGx_2024.html">Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance</a></li>
    <li><a class="dropdown-item" href="../articles/Workshop_Outline.html">Workshop Outline and Objectives</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Module Labs</h6></li>
    <li><a class="dropdown-item" href="../articles/Module1.html">Module 1: Introduction to pharmacogenomics</a></li>
    <li><a class="dropdown-item" href="../articles/Module2.html">Module 2: Hands-on with pharmacogenomics data</a></li>
    <li><a class="dropdown-item" href="../articles/Module3.html">Module 3: Pharmacogenomics for biomarker discovery – Basic analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Module4.html">Module 4: Pharmacogenomics for biomarker discovery - Advanced analysis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/bhklab/CBWWorkshop2024"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Module 3: Pharmacogenomics for biomarker discovery – Basic analysis</h1>
                        <h4 data-toc-skip class="author">Jermiah J.
Joseph</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
      </address>
                              <h4 data-toc-skip class="author">Nikta
Feizi</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:nikta.feizi@uhn.ca" class="email">nikta.feizi@uhn.ca</a>
      </address>
                              <h4 data-toc-skip class="author">Julia
Nguyen</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a>
      </address>
                  
            <h4 data-toc-skip class="date">18 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/bhklab/CBWWorkshop2024/blob/main/vignettes/Module3.Rmd" class="external-link"><code>vignettes/Module3.Rmd</code></a></small>
      <div class="d-none name"><code>Module3.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
</div>
<div class="section level2">
<h2 class="unnumbered" id="lab-3-overview">Lab 3 Overview<a class="anchor" aria-label="anchor" href="#lab-3-overview"></a>
</h2>
<div class="section level3">
<h3 class="unnumbered" id="instructors-names-and-contact-information">Instructor(s) name(s) and contact
information<a class="anchor" aria-label="anchor" href="#instructors-names-and-contact-information"></a>
</h3>
<ul>
<li>Jermiah J. Joseph <a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
</li>
<li>Nikta Feizi &lt;&gt;</li>
<li>Julia Nguyen <a href="mailto:julia.nguyen@uhn.ca" class="email">julia.nguyen@uhn.ca</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="pharmacogx-package-overview">PharmacoGx Package Overview<a class="anchor" aria-label="anchor" href="#pharmacogx-package-overview"></a>
</h3>
<p><strong>PharmacoGx:</strong> An R package for analyzing
pharmacogenomic datasets.</p>
<p><strong>Key Functions:</strong></p>
<ul>
<li>
<p><strong>Install and Load PharmacoGx</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"PharmacoGx"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">PharmacoGx</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p><strong>Download a PharmacoSet</strong></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/availablePSets.html" class="external-link">availablePSets</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">GDSC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html" class="external-link">downloadPSet</a></span><span class="op">(</span><span class="st">"GDSC_2020(v2-8.2)"</span><span class="op">)</span></span>
<span><span class="co"># downloadPSet(): Download pharmacogenomic datasets.</span></span></code></pre></div>
</li>
<li>
<p><strong>Extract Drug Response Data</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">drug_response</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeSensitivityProfiles.html" class="external-link">summarizeSensitivityProfiles</a></span><span class="op">(</span><span class="va">GDSC</span>, sensitivity.measure<span class="op">=</span><span class="st">'aac_recomputed'</span><span class="op">)</span></span>
<span><span class="co"># summarizeSensitivityProfiles(): Summarize drug response data.</span></span></code></pre></div>
</li>
<li>
<p><strong>Extract Gene Expression Data</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeMolecularProfiles.html" class="external-link">summarizeMolecularProfiles</a></span><span class="op">(</span><span class="va">GDSC</span>, mDataType<span class="op">=</span><span class="st">'rna'</span><span class="op">)</span> </span>
<span><span class="va">gene_expr_mtx</span> <span class="op">&lt;-</span> <span class="fu">assay</span><span class="op">(</span><span class="va">gene_expr</span><span class="op">)</span></span></code></pre></div>
</li>
</ul>
<p><strong>Data Retrieval:</strong></p>
<ul>
<li><p><strong>availablePSets()</strong>: Lists available
pharmacogenomic datasets.</p></li>
<li>
<p><strong>Downloading Multiple PharmacoSets:</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PSet.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GDSC"</span>, <span class="st">"CCLE"</span>, <span class="st">"CTRP"</span><span class="op">)</span>, <span class="va">downloadPSet</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PSet.list</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GDSC"</span>, <span class="st">"CCLE"</span>, <span class="st">"CTRP"</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p><strong>Accessing Molecular Profiles:</strong></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">molecular.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">PSet.list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">pset</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeMolecularProfiles.html" class="external-link">summarizeMolecularProfiles</a></span><span class="op">(</span><span class="va">pset</span>, mDataType <span class="op">=</span> <span class="st">'rna'</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>In this workshop, we will explore data integration and comparative
analysis using the <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
package, focusing on the <strong>GDSC</strong> and <strong>CCLE</strong>
datasets. Specifically, we will cover drug sensitivity comparisons using
AUC and IC50 measures and investigate correlations between gene
expression profiles. This R Markdown file is designed to give
participants hands-on experience with data exploration and
interpretation.</p>
<div class="section level4">
<h4 id="learning-objectives">Learning Objectives<a class="anchor" aria-label="anchor" href="#learning-objectives"></a>
</h4>
<ol style="list-style-type: decimal">
<li>Learn how to load and handle data from the <strong>GDSC</strong> and
<strong>CCLE</strong> datasets.</li>
<li>Understand how to summarize drug sensitivity profiles and molecular
data.</li>
<li>Perform correlations to assess concordance between cell line
datasets.</li>
<li>Visualize and interpret the results of statistical tests.</li>
</ol>
</div>
</div>
<div class="section level3">
<h3 id="exploring-gdsc-and-ccle-datasets">Exploring GDSC and CCLE Datasets<a class="anchor" aria-label="anchor" href="#exploring-gdsc-and-ccle-datasets"></a>
</h3>
<div class="section level4">
<h4 id="load-gdsc-and-ccle-datasets">Load GDSC and CCLE Datasets<a class="anchor" aria-label="anchor" href="#load-gdsc-and-ccle-datasets"></a>
</h4>
<p>We will use the <strong>GDSCsmall</strong> and
<strong>CCLEsmall</strong> sample datasets provided by <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
to compare the two data sources.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"GDSCsmall"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CCLEsmall"</span><span class="op">)</span></span>
<span><span class="va">GDSCsmall</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">GDSC</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Mon Aug 24 15:18:23 2015</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">10</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>RNA :</span></span>
<span><span class="co">##   Dim: 300, 11 </span></span>
<span><span class="co">## rna2 :</span></span>
<span><span class="co">##   Dim: 300, 9 </span></span>
<span><span class="co">## mutation :</span></span>
<span><span class="co">##   Dim: 70, 10 </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: </span>Drug pertubation:</span></span>
<span><span class="co">##    Please look at pertNumber(cSet) to determine number of experiments  for each drug-sample combination.</span></span>
<span><span class="co">## Drug sensitivity:</span></span>
<span><span class="co">##    Number of Experiments:  972 </span></span>
<span><span class="co">##    Please look at sensNumber(cSet) to determine number of  experiments for each drug-sample combination.</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="find-common-genes-between-gdsc-and-ccle">Find Common Genes Between GDSC and CCLE<a class="anchor" aria-label="anchor" href="#find-common-genes-between-gdsc-and-ccle"></a>
</h4>
<p>The first step in comparing datasets is to find shared features
between them. Here, we identify the genes that are present in both
datasets.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">commonGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">fNames</a></span><span class="op">(</span><span class="va">GDSCsmall</span>, <span class="st">"rna"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">fNames</a></span><span class="op">(</span><span class="va">CCLEsmall</span>, <span class="st">"rna"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">commonGenes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 41</span></span></code></pre>
<p>There are 41 genes that are common between the RNA profiles of the
GDSC and CCLE datasets.</p>
</div>
<div class="section level4">
<h4 id="identify-common-cell-lines-and-drugs">Identify Common Cell Lines and Drugs<a class="anchor" aria-label="anchor" href="#identify-common-cell-lines-and-drugs"></a>
</h4>
<p>Next, we identify the common cell lines and drugs between
<strong>GDSC</strong> and <strong>CCLE</strong>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">common</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/intersectPSet.html" class="external-link">intersectPSet</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"CCLE"</span> <span class="op">=</span> <span class="va">CCLEsmall</span>, <span class="st">"GDSC"</span> <span class="op">=</span> <span class="va">GDSCsmall</span><span class="op">)</span>,</span>
<span>  intersectOn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cell.lines"</span>, <span class="st">"drugs"</span><span class="op">)</span>,</span>
<span>  strictIntersect <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Intersecting large PSets may take a long time ...</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">cellNames</a></span><span class="op">(</span><span class="va">common</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "22RV1"    "23132-87" "5637"     "639-V"    "647-V"    "697"      "769-P"   </span></span>
<span><span class="co">## [8] "786-0"    "8-MG-BA"</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">cellNames</a></span><span class="op">(</span><span class="va">common</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 9</span></span></code></pre>
<p>There are 9 common samples between GDSC and CCLE. The
<code>intersectPSet</code> function returns a list of the two PSets
subsetted to include only the common cell lines.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">common</span></span></code></pre></div>
<pre><code><span><span class="co">## $CCLE</span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">CCLE</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Fri Nov  6 14:00:53 2015</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">9</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>RNA :</span></span>
<span><span class="co">##   Dim: 50, 9 </span></span>
<span><span class="co">## RNAseq :</span></span>
<span><span class="co">##   Dim: 50, 9 </span></span>
<span><span class="co">## mutation :</span></span>
<span><span class="co">##   Dim: 1667, 9 </span></span>
<span><span class="co">## CNV :</span></span>
<span><span class="co">##   Dim: 50, 7 </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: </span>Drug pertubation:</span></span>
<span><span class="co">##    Please look at pertNumber(cSet) to determine number of experiments  for each drug-sample combination.</span></span>
<span><span class="co">## Drug sensitivity:</span></span>
<span><span class="co">##    Number of Experiments:  102 </span></span>
<span><span class="co">##    Please look at sensNumber(cSet) to determine number of  experiments for each drug-sample combination.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $GDSC</span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">&lt;PharmacoSet&gt;</span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## Name: </span><span style="color: #00BB00;">GDSC</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Date Created: </span><span style="color: #00BB00;">Mon Aug 24 15:18:23 2015</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Number of samples: </span> <span style="color: #00BB00;">9</span> </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Molecular profiles: </span></span></span>
<span><span class="co"><span style="color: #BBBB00; font-weight: bold; font-style: italic;">## </span>RNA :</span></span>
<span><span class="co">##   Dim: 300, 10 </span></span>
<span><span class="co">## rna2 :</span></span>
<span><span class="co">##   Dim: 300, 9 </span></span>
<span><span class="co">## mutation :</span></span>
<span><span class="co">##   Dim: 70, 9 </span></span>
<span><span class="co">## <span style="color: #BBBB00; font-weight: bold; font-style: italic;">Treatment response: </span>Drug pertubation:</span></span>
<span><span class="co">##    Please look at pertNumber(cSet) to determine number of experiments  for each drug-sample combination.</span></span>
<span><span class="co">## Drug sensitivity:</span></span>
<span><span class="co">##    Number of Experiments:  79 </span></span>
<span><span class="co">##    Please look at sensNumber(cSet) to determine number of  experiments for each drug-sample combination.</span></span></code></pre>
<p>Notice that most of the molecular profiles are subsetted to include
the 9 common samples. Some odd cases include RNA from GDSC which has a
technical replicate (hence 10 samples) and CNV from CCLE which is
missing CNV profiles for two of the nine cell lines.</p>
</div>
<div class="section level4">
<h4 id="summarize-drug-sensitivity-profiles">Summarize Drug Sensitivity Profiles<a class="anchor" aria-label="anchor" href="#summarize-drug-sensitivity-profiles"></a>
</h4>
<p>We summarize drug sensitivity profiles (AUC and IC50) for each
dataset. The <code>summary.stat</code> parameter can be set to different
statistical metrics such as <code>mean</code>, <code>median</code>, etc.
Here, we use the <code>median</code> for summarizing the
sensitivity.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Summary statistics for AUC</span></span>
<span><span class="va">GDSC.auc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeSensitivityProfiles.html" class="external-link">summarizeSensitivityProfiles</a></span><span class="op">(</span></span>
<span>  <span class="va">common</span><span class="op">$</span><span class="va">GDSC</span>,</span>
<span>  sensitivity.measure <span class="op">=</span> <span class="st">"auc_published"</span>,</span>
<span>  summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">CCLE.auc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeSensitivityProfiles.html" class="external-link">summarizeSensitivityProfiles</a></span><span class="op">(</span></span>
<span>  <span class="va">common</span><span class="op">$</span><span class="va">CCLE</span>,</span>
<span>  sensitivity.measure <span class="op">=</span> <span class="st">"auc_published"</span>,</span>
<span>  summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summary statistics for IC50</span></span>
<span><span class="va">GDSC.ic50</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeSensitivityProfiles.html" class="external-link">summarizeSensitivityProfiles</a></span><span class="op">(</span></span>
<span>  <span class="va">common</span><span class="op">$</span><span class="va">GDSC</span>,</span>
<span>  sensitivity.measure <span class="op">=</span> <span class="st">"ic50_published"</span>,</span>
<span>  summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">CCLE.ic50</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeSensitivityProfiles.html" class="external-link">summarizeSensitivityProfiles</a></span><span class="op">(</span></span>
<span>  <span class="va">common</span><span class="op">$</span><span class="va">CCLE</span>,</span>
<span>  sensitivity.measure <span class="op">=</span> <span class="st">"ic50_published"</span>,</span>
<span>  summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">CCLE.auc</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##               22RV1 23132-87      5637     639-V 647-V       697      769-P</span></span>
<span><span class="co">## PD-0325901 0.385000       NA 0.1708125 0.3088750    NA 0.2291000 0.23250000</span></span>
<span><span class="co">## 17-AAG     0.372460       NA 0.4828250 0.4877500    NA 0.3420875 0.40937500</span></span>
<span><span class="co">## Nilotinib  0.000000       NA 0.0072625 0.0710125    NA 0.1573375 0.00000000</span></span>
<span><span class="co">## PHA-665752 0.094375       NA 0.0000000 0.0194000    NA 0.1609750 0.09427143</span></span>
<span><span class="co">## lapatinib  0.036125       NA 0.2212250 0.0066000    NA 0.0453250 0.16475000</span></span>
<span><span class="co">## Nutlin-3   0.076925       NA 0.0000000 0.0666625    NA 0.2588125 0.08598571</span></span>
<span><span class="co">##                 786-0   8-MG-BA</span></span>
<span><span class="co">## PD-0325901 0.13862500 0.0758750</span></span>
<span><span class="co">## 17-AAG     0.44162500 0.4426500</span></span>
<span><span class="co">## Nilotinib  0.07501250        NA</span></span>
<span><span class="co">## PHA-665752 0.07408625 0.3962500</span></span>
<span><span class="co">## lapatinib  0.12987500 0.0547625</span></span>
<span><span class="co">## Nutlin-3   0.00000000 0.0655125</span></span></code></pre>
<p>We can visualize the distribution of drug sensitivity profiles to
compare between the two psets.</p>
<p>First we quickly reformat the data into a long format to facilitate
plotting. We use the <code>melt()</code> function from the
<code>reshape2</code> package.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CCLE_toPlot</span> <span class="op">&lt;-</span> <span class="fu">melt</span><span class="op">(</span><span class="va">CCLE.auc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">CCLE_toPlot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Drug"</span>, <span class="st">"Gene"</span>, <span class="st">"AUC"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">CCLE_toPlot</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         Drug  Gene      AUC</span></span>
<span><span class="co">## 1 PD-0325901 22RV1 0.385000</span></span>
<span><span class="co">## 2     17-AAG 22RV1 0.372460</span></span>
<span><span class="co">## 3  Nilotinib 22RV1 0.000000</span></span>
<span><span class="co">## 4 PHA-665752 22RV1 0.094375</span></span>
<span><span class="co">## 5  lapatinib 22RV1 0.036125</span></span>
<span><span class="co">## 6   Nutlin-3 22RV1 0.076925</span></span></code></pre>
<p>Next, we can create a quick box plot to visualize the AUC
distributions per drug.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot</span><span class="op">(</span><span class="va">CCLE_toPlot</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Drug</span>, y <span class="op">=</span> <span class="va">AUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_boxplot</span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"grey"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="co">#rotate x-axis labels</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 33 rows containing non-finite outside the scale range</span></span>
<span><span class="co">## (`stat_boxplot()`).</span></span></code></pre>
<p><img src="Module3_files/figure-html/plot_sensitivity-1.png" width="700"></p>
<p>Let’s do this again but with the GDSC PSet so we can compare the
two</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># format the CCLE sensitivity data</span></span>
<span><span class="va">GDSC_toPlot</span> <span class="op">&lt;-</span> <span class="fu">melt</span><span class="op">(</span><span class="va">GDSC.auc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">GDSC_toPlot</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Drug"</span>, <span class="st">"Gene"</span>, <span class="st">"AUC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># merge the two pset dataframes</span></span>
<span><span class="va">GDSC_toPlot</span><span class="op">$</span><span class="va">PSet</span> <span class="op">&lt;-</span> <span class="st">"GDSC"</span></span>
<span><span class="va">CCLE_toPlot</span><span class="op">$</span><span class="va">PSet</span> <span class="op">&lt;-</span> <span class="st">"CCLE"</span></span>
<span><span class="va">merge_toPlot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">GDSC_toPlot</span>, <span class="va">CCLE_toPlot</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot to compare AUC distribution between PSets</span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">merge_toPlot</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Drug</span>, y <span class="op">=</span> <span class="va">AUC</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_boxplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>fill <span class="op">=</span> <span class="va">PSet</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#624763"</span>, <span class="st">"#B1D3A3"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme</span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="co">#rotate x-axis labels</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 89 rows containing non-finite outside the scale range</span></span>
<span><span class="co">## (`stat_boxplot()`).</span></span></code></pre>
<p><img src="Module3_files/figure-html/plot_sensitivity_compare-1.png" width="700"></p>
<p>Preclinical drug response data is fairly noisy, hence the variation
we see between the two datasets. However, we do see some commonalities
in distribution.</p>
</div>
<div class="section level4">
<h4 id="summarize-gene-expression-profiles">Summarize Gene Expression Profiles<a class="anchor" aria-label="anchor" href="#summarize-gene-expression-profiles"></a>
</h4>
<p>We summarize gene expression data for the genes that are common
between <strong>GDSC</strong> and <strong>CCLE</strong>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GDSCexpression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeMolecularProfiles.html" class="external-link">summarizeMolecularProfiles</a></span><span class="op">(</span></span>
<span>  <span class="va">common</span><span class="op">$</span><span class="va">GDSC</span>, <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">cellNames</a></span><span class="op">(</span><span class="va">common</span><span class="op">$</span><span class="va">GDSC</span><span class="op">)</span>,</span>
<span>  mDataType <span class="op">=</span> <span class="st">"rna"</span>, features <span class="op">=</span> <span class="va">commonGenes</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">assay</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">CCLEexpression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/summarizeMolecularProfiles.html" class="external-link">summarizeMolecularProfiles</a></span><span class="op">(</span></span>
<span>  <span class="va">common</span><span class="op">$</span><span class="va">CCLE</span>, <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">cellNames</a></span><span class="op">(</span><span class="va">common</span><span class="op">$</span><span class="va">CCLE</span><span class="op">)</span>,</span>
<span>  mDataType <span class="op">=</span> <span class="st">"rna"</span>, features <span class="op">=</span> <span class="va">commonGenes</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">assay</span><span class="op">(</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">GDSCexpression</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                     22RV1  23132-87      5637     639-V     647-V       697</span></span>
<span><span class="co">## ENSG00000000003  5.890740  6.243345 10.018953  9.653157 10.039302  5.215355</span></span>
<span><span class="co">## ENSG00000000005  4.448758  4.335738  4.108065  4.014940  4.250878  4.367339</span></span>
<span><span class="co">## ENSG00000000419 10.427302 11.045576 11.611295 11.604278 11.248686 10.572231</span></span>
<span><span class="co">## ENSG00000000457  5.905884  6.900556  5.684007  5.392485  5.748209  5.934834</span></span>
<span><span class="co">## ENSG00000000460  6.010605  5.433439  5.729120  5.439058  6.453635  6.342021</span></span>
<span><span class="co">## ENSG00000000938  4.410785  4.329849  4.397001  4.176387  4.275824  4.618786</span></span>
<span><span class="co">##                    769-P     786-0   8-MG-BA</span></span>
<span><span class="co">## ENSG00000000003 8.615450  7.304893  7.994002</span></span>
<span><span class="co">## ENSG00000000005 4.722970  4.235424  4.447925</span></span>
<span><span class="co">## ENSG00000000419 9.927782 10.825907 10.759706</span></span>
<span><span class="co">## ENSG00000000457 6.475836  5.087249  4.994176</span></span>
<span><span class="co">## ENSG00000000460 5.999142  5.001772  6.073944</span></span>
<span><span class="co">## ENSG00000000938 5.003250  4.443563  4.797094</span></span></code></pre>
<p>Here, you could do PCA on the profiles to quickly check the data. We
will skip this step in the interest of time.</p>
</div>
</div>
<div class="section level3">
<h3 id="correlation-analysis-between-gdsc-and-ccle">Correlation Analysis Between GDSC and CCLE<a class="anchor" aria-label="anchor" href="#correlation-analysis-between-gdsc-and-ccle"></a>
</h3>
<p>We perform correlation analysis to examine the relationship between
gene expression, AUC, and IC50 measures across the <strong>GDSC</strong>
and <strong>CCLE</strong> datasets using <strong>Spearman’s
correlation</strong>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get common cell line names</span></span>
<span><span class="va">cc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/CoreSet-accessors.html" class="external-link">cellNames</a></span><span class="op">(</span><span class="va">common</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># correlation of gene expression across common cell lines</span></span>
<span><span class="va">ge.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">cc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">d1</span>, <span class="va">d2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span></span>
<span>    <span class="va">d1</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span>, <span class="va">d2</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"spearman"</span>, use <span class="op">=</span> <span class="st">"pairwise.complete.obs"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span>, d1 <span class="op">=</span> <span class="va">GDSCexpression</span>, d2 <span class="op">=</span> <span class="va">CCLEexpression</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ge.cor</span></span></code></pre></div>
<pre><code><span><span class="co">##     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span><span class="co">## 0.8658537 0.8844948 0.8935540 0.9202091 0.9106272 0.8893728 0.8939024 0.8681185 </span></span>
<span><span class="co">##   8-MG-BA </span></span>
<span><span class="co">## 0.8470383</span></span></code></pre>
<p>We can see that the RNA-Seq expression values are highly correlated
across the common cell lines.</p>
<p>Let’s take a look at the correlation of drug response, starting with
IC50 values.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># quick look at the CCLE IC50 data</span></span>
<span><span class="va">CCLE.ic50</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                22RV1 23132-87       5637     639-V 647-V       697     769-P</span></span>
<span><span class="co">## PD-0325901 8.0000000       NA 8.00000000 8.0000000    NA 8.0000000 1.4542717</span></span>
<span><span class="co">## 17-AAG     0.3297017       NA 0.07082279 0.1500945    NA 0.4225712 0.1517988</span></span>
<span><span class="co">## Nilotinib  8.0000000       NA 7.47535467 8.0000000    NA 1.9104344 8.0000000</span></span>
<span><span class="co">## PHA-665752 8.0000000       NA 8.00000000 8.0000000    NA 3.2754419 7.1504111</span></span>
<span><span class="co">## lapatinib  7.8473053       NA 2.30776763 8.0000000    NA 0.8494762 1.0574611</span></span>
<span><span class="co">## Nutlin-3   8.0000000       NA 8.00000000 8.0000000    NA 1.8905237 5.5913019</span></span>
<span><span class="co">##                786-0   8-MG-BA</span></span>
<span><span class="co">## PD-0325901 8.0000000 8.0000000</span></span>
<span><span class="co">## 17-AAG     0.2795302 0.2349526</span></span>
<span><span class="co">## Nilotinib  8.0000000        NA</span></span>
<span><span class="co">## PHA-665752 8.0000000 8.0000000</span></span>
<span><span class="co">## lapatinib  7.1780353 5.2266199</span></span>
<span><span class="co">## Nutlin-3   8.0000000 8.0000000</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># correlation of IC50 values across common drugs and cell lines</span></span>
<span><span class="va">ic50.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">cc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">d1</span>, <span class="va">d2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span></span>
<span>    <span class="va">d1</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span>, <span class="va">d2</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"spearman"</span>, use <span class="op">=</span> <span class="st">"pairwise.complete.obs"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span>, d1 <span class="op">=</span> <span class="va">GDSC.ic50</span>, d2 <span class="op">=</span> <span class="va">CCLE.ic50</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ic50.cor</span></span></code></pre></div>
<pre><code><span><span class="co">##     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span><span class="co">## 0.4082483        NA 0.5713800 0.4082483        NA 0.6273461 0.6300619 0.6546537 </span></span>
<span><span class="co">##   8-MG-BA </span></span>
<span><span class="co">## 0.3664850</span></span></code></pre>
<p>Notice that we have some <code>NA</code> values in the original CCLE
IC50 dataset and in the correlation results.</p>
<p>The <code>pairwise.complete.obs</code> argument used in the
<code><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor()</a></code> function ensures that only complete observations
(i.e. numeric values that are not <code>NA</code>) are included in the
pairwise correlations.</p>
<p>Since <code>23132-87</code> and <code>647-V</code> are all
<code>NA</code> values, no spearman correlation was computed.</p>
<p>We can quickly compute the correlation for AUC as well.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># correlation of AUC values across common drugs and cell lines</span></span>
<span><span class="va">auc.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">cc</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">d1</span>, <span class="va">d2</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span></span>
<span>    <span class="va">d1</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span>, <span class="va">d2</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"spearman"</span>, use <span class="op">=</span> <span class="st">"pairwise.complete.obs"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span>, d1 <span class="op">=</span> <span class="va">GDSC.auc</span>, d2 <span class="op">=</span> <span class="va">CCLE.auc</span><span class="op">)</span></span>
<span></span>
<span><span class="va">auc.cor</span></span></code></pre></div>
<pre><code><span><span class="co">##     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span><span class="co">## 0.0000000        NA 0.7748062 0.8214286        NA 0.3750000 0.8571429 0.4638168 </span></span>
<span><span class="co">##   8-MG-BA </span></span>
<span><span class="co">## 0.5414447</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="statistical-comparison">Statistical Comparison<a class="anchor" aria-label="anchor" href="#statistical-comparison"></a>
</h3>
<p>We compare the correlations using <strong>Wilcoxon signed-rank
tests</strong> to see if there are significant differences between the
gene expression correlations and the drug sensitivity correlations.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w1</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html" class="external-link">wilcox.test</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">ge.cor</span>, y <span class="op">=</span> <span class="va">auc.cor</span>, conf.int <span class="op">=</span> <span class="cn">TRUE</span>, exact <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">w2</span> <span class="op">&lt;-</span> <span class="fu">stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/stats/wilcox.test.html" class="external-link">wilcox.test</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">ge.cor</span>, y <span class="op">=</span> <span class="va">ic50.cor</span>, conf.int <span class="op">=</span> <span class="cn">TRUE</span>, exact <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">w1</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  ge.cor and auc.cor</span></span>
<span><span class="co">## W = 62, p-value = 0.001496</span></span>
<span><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span><span class="co">## 95 percent confidence interval:</span></span>
<span><span class="co">##  0.06309566 0.51857816</span></span>
<span><span class="co">## sample estimates:</span></span>
<span><span class="co">## difference in location </span></span>
<span><span class="co">##               0.347967</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">w2</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Wilcoxon rank sum test with continuity correction</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## data:  ge.cor and ic50.cor</span></span>
<span><span class="co">## W = 63, p-value = 0.001024</span></span>
<span><span class="co">## alternative hypothesis: true location shift is not equal to 0</span></span>
<span><span class="co">## 95 percent confidence interval:</span></span>
<span><span class="co">##  0.2407670 0.4857055</span></span>
<span><span class="co">## sample estimates:</span></span>
<span><span class="co">## difference in location </span></span>
<span><span class="co">##              0.3179657</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Display p-values</span></span>
<span><span class="va">ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"GE vs. AUC = %.1E\nGE vs. IC50 = %.1E"</span>, <span class="va">w1</span><span class="op">$</span><span class="va">p.value</span>, <span class="va">w2</span><span class="op">$</span><span class="va">p.value</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">ss</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GE vs. AUC = 1.5E-03</span></span>
<span><span class="co">## GE vs. IC50 = 1.0E-03</span></span></code></pre>
<div class="section level4">
<h4 id="boxplot-visualization">Boxplot Visualization<a class="anchor" aria-label="anchor" href="#boxplot-visualization"></a>
</h4>
<p>The results are visualized using boxplots to compare the correlations
across gene expression, AUC, and IC50.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"GE"</span> <span class="op">=</span> <span class="va">ge.cor</span>, <span class="st">"AUC"</span> <span class="op">=</span> <span class="va">auc.cor</span>, <span class="st">"IC50"</span> <span class="op">=</span> <span class="va">ic50.cor</span><span class="op">)</span>,</span>
<span>  main <span class="op">=</span> <span class="st">"Concordance between cell lines"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">R</span><span class="op">[</span><span class="va">s</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  sub <span class="op">=</span> <span class="va">ss</span>,</span>
<span>  ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>  col <span class="op">=</span> <span class="st">"lightgrey"</span>,</span>
<span>  pch <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  border <span class="op">=</span> <span class="st">"black"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="Module3_files/figure-html/boxplot_results-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="statistical-analysis-for-drug-response-associations">Statistical Analysis for Drug Response Associations<a class="anchor" aria-label="anchor" href="#statistical-analysis-for-drug-response-associations"></a>
</h3>
<p>In this final section, we will be focusing on computing the
association between a feature of interest and drug response in order to
identify predictive biomarkers.</p>
<p>Let’s begin by selecting a feature and drug of interest to
explore.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">commonGenes</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSG00000000003" "ENSG00000000005" "ENSG00000000419" "ENSG00000000457"</span></span>
<span><span class="co">## [5] "ENSG00000000460" "ENSG00000000938"</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">feature</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000000003"</span></span>
<span></span>
<span><span class="va">common</span><span class="op">$</span><span class="va">GDSC</span><span class="op">@</span><span class="va">treatment</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "PD-0325901" "17-AAG"     "Nilotinib"  "PHA-665752" "lapatinib" </span></span>
<span><span class="co">##  [6] "Nutlin-3"   "AZD0530"    "Crizotinib" "Sorafenib"  "PD-0332991"</span></span>
<span><span class="co">## [11] "paclitaxel" "AZD6244"    "PLX4720"    "TAE684"     "Erlotinib"</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">drug</span> <span class="op">&lt;-</span> <span class="st">"PD-0325901"</span></span></code></pre></div>
<p>Next, we need to extract the vector of gene expression for our chosen
feature and drug response for our chosen drug. Let’s do this for the
GDSC dataset.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Gene</span> <span class="op">&lt;-</span> <span class="va">GDSCexpression</span><span class="op">[</span><span class="va">feature</span>,<span class="op">]</span></span>
<span><span class="va">Drug</span> <span class="op">&lt;-</span> <span class="va">GDSC.auc</span><span class="op">[</span><span class="va">drug</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">Gene</span></span></code></pre></div>
<pre><code><span><span class="co">##     22RV1  23132-87      5637     639-V     647-V       697     769-P     786-0 </span></span>
<span><span class="co">##  5.890740  6.243345 10.018953  9.653157 10.039302  5.215355  8.615450  7.304893 </span></span>
<span><span class="co">##   8-MG-BA </span></span>
<span><span class="co">##  7.994002</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Drug</span></span></code></pre></div>
<pre><code><span><span class="co">##    22RV1 23132-87     5637    639-V    647-V      697    769-P    786-0 </span></span>
<span><span class="co">## 0.029649 0.122911 0.007242 0.279261 0.100351 0.049681 0.179831 0.150231 </span></span>
<span><span class="co">##  8-MG-BA </span></span>
<span><span class="co">## 0.034754</span></span></code></pre>
<p>Notice that the PSet has already kept the samples ordered across the
various profiles.</p>
<p>Before we compute the associations, we can quickly plot the two
variables to visualize their relationship.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create datafarme to plot</span></span>
<span><span class="va">toPlot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Gene <span class="op">=</span> <span class="va">Gene</span>, Drug <span class="op">=</span> <span class="va">Drug</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">ggplot</span><span class="op">(</span><span class="va">toPlot</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">Gene</span>, y <span class="op">=</span> <span class="va">Drug</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_point</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme_classic</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Gene (expr)"</span>, y <span class="op">=</span> <span class="st">"Drug (AUC)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Module3_files/figure-html/plot_association-1.png" width="700"></p>
<p>From the scatter plot, we see a weak positive correlation. Let’s see
if this is reflected in our association analyses.</p>
<div class="section level4">
<h4 id="concordance-index">Concordance Index<a class="anchor" aria-label="anchor" href="#concordance-index"></a>
</h4>
<p>Fist, we will compute the association between the gene expression
data and drug response. We will use the <code>concordance.index</code>
function from the <code>survcomp</code> package.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ci</span> <span class="op">&lt;-</span> <span class="fu">survcomp</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/survcomp/man/concordance.index.html" class="external-link">concordance.index</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">Drug</span><span class="op">)</span>,                   <span class="co"># drug vector</span></span>
<span>  surv.time <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span>,       <span class="co"># gene vector</span></span>
<span>  surv.event <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span><span class="op">)</span>,   </span>
<span>  outx <span class="op">=</span> <span class="cn">TRUE</span>, method<span class="op">=</span><span class="st">"noether"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Concordance Index:"</span>, <span class="va">ci</span><span class="op">$</span><span class="va">c.index</span>, <span class="st">"\n"</span>,</span>
<span>    <span class="st">"P-value:"</span>, <span class="va">ci</span><span class="op">$</span><span class="va">p.value</span>, <span class="st">"\n"</span>,</span>
<span>    <span class="st">"Standard Error:"</span>, <span class="va">ci</span><span class="op">$</span><span class="va">se</span>, <span class="st">"\n"</span>, </span>
<span>    <span class="st">"Upper CI:"</span>, <span class="va">ci</span><span class="op">$</span><span class="va">upper</span>,<span class="st">"\n"</span>,</span>
<span>    <span class="st">"Lower CI:"</span>, <span class="va">ci</span><span class="op">$</span><span class="va">lower</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Concordance Index: 0.4166667 </span></span>
<span><span class="co">##  P-value: 0.07592696 </span></span>
<span><span class="co">##  Standard Error: 0.04695301 </span></span>
<span><span class="co">##  Upper CI: 0.5086929 </span></span>
<span><span class="co">##  Lower CI: 0.3246404</span></span></code></pre>
<p>The concordance index tells us there is a weak inverse association
between the gene expression and the drug response.</p>
</div>
<div class="section level4">
<h4 id="pearsons-correlation-coefficient">Pearson’s Correlation Coefficient<a class="anchor" aria-label="anchor" href="#pearsons-correlation-coefficient"></a>
</h4>
<p>Using the same feature and drug, let’s compute the association again
but this time using Pearson’s correlation.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">Gene</span>,</span>
<span>  y <span class="op">=</span> <span class="va">Drug</span>,</span>
<span>  alternative <span class="op">=</span> <span class="st">"two.sided"</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"pearson"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Pearson's correlation:"</span>, <span class="va">ps</span><span class="op">$</span><span class="va">estimate</span>, <span class="st">"\n"</span>,</span>
<span>    <span class="st">"P-value:"</span>, <span class="va">ci</span><span class="op">$</span><span class="va">p.value</span>, <span class="st">"\n"</span>,</span>
<span>    <span class="st">"Confidence Interval:"</span>, <span class="va">ps</span><span class="op">$</span><span class="va">conf.int</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Pearson's correlation: 0.2937677 </span></span>
<span><span class="co">##  P-value: 0.07592696 </span></span>
<span><span class="co">##  Confidence Interval: -0.4601228 0.8015157</span></span></code></pre>
<p>The Pearson’s correlation tells us there is a (very) weak positive
linear relationship between the gene expression and the drug
response.</p>
<p>Notice the slight difference in interpretation between the
concordance index and the Pearson’s correlation.</p>
</div>
<div class="section level4">
<h4 id="drug-sensitivity-signatures-pharmacogx">Drug Sensitivity Signatures (PharmacoGx)<a class="anchor" aria-label="anchor" href="#drug-sensitivity-signatures-pharmacogx"></a>
</h4>
<p>The last method we will explore is a built-in function from the
<code>PharmacoGx</code> package called <code>drugSensitivitySig</code>.
This function takes a <code>PharmacoSet</code> along with a list of
drugs and features, then computes the association between the feature
expression and drug response for each pair.</p>
<p>Let’s first select the first 3 genes and drugs as our features &amp;
drugs of interest:</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">features</span> <span class="op">&lt;-</span> <span class="va">commonGenes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="va">drugs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">common</span><span class="op">$</span><span class="va">GDSC</span><span class="op">@</span><span class="va">treatment</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span></span>
<span><span class="va">features</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSG00000000003" "ENSG00000000005" "ENSG00000000419"</span></span></code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">drugs</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "PD-0325901" "17-AAG"     "Nilotinib"</span></span></code></pre>
<p>Now we can calculate the drug sensitivity signature using the vector
of genes and drugs for the GDSC dataset. Note that this gene-drug
signature is based on univariable analysis.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># gene expression signature</span></span>
<span><span class="va">sig.rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/drugSensitivitySig.html" class="external-link">drugSensitivitySig</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">GDSCsmall</span>,</span>
<span>  mDataType <span class="op">=</span> <span class="st">"rna"</span>,</span>
<span>  drugs <span class="op">=</span> <span class="va">drugs</span>,</span>
<span>  features <span class="op">=</span> <span class="va">features</span>,</span>
<span>  sensitivity.measure <span class="op">=</span> <span class="st">"auc_published"</span>,</span>
<span>  molecular.summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  sensitivity.summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  modeling.method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">sig.rna</span><span class="op">@</span><span class="va">.Data</span><span class="op">[</span>,,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## , , estimate</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                  PD-0325901     17-AAG  Nilotinib</span></span>
<span><span class="co">## ENSG00000000003 -0.18840373 -0.2369481  0.8554590</span></span>
<span><span class="co">## ENSG00000000005 -0.14409315 -0.4756331  0.7875267</span></span>
<span><span class="co">## ENSG00000000419 -0.02890968  0.5466780 -0.8239237</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## , , pvalue</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                 PD-0325901    17-AAG Nilotinib</span></span>
<span><span class="co">## ENSG00000000003  0.8793373 0.8477058 0.3465502</span></span>
<span><span class="co">## ENSG00000000005  0.9079470 0.6844380 0.4227225</span></span>
<span><span class="co">## ENSG00000000419  0.9815930 0.6317843 0.3835609</span></span></code></pre>
<p>The output is a 3D array of genes x drugs x metric. Outputted are the
<code>estimate</code> and <code>pvalue</code>. Feel free to explore the
other array components in <code>sig.rna@.Data</code>.</p>
<p>This function is a powerful method for investigating the univariate
relationship between multiple molecular features and drugs.</p>
<p>Let’s do the same for mutations, we’ll do a quick example using one
gene and drug from the CCLE dataset.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># mutation signature</span></span>
<span><span class="va">sig.mut</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CoreGx/man/drugSensitivitySig.html" class="external-link">drugSensitivitySig</a></span><span class="op">(</span></span>
<span>  object <span class="op">=</span> <span class="va">CCLEsmall</span>,</span>
<span>  mDataType <span class="op">=</span> <span class="st">"mutation"</span>,</span>
<span>  drugs <span class="op">=</span> <span class="st">"PD-0325901"</span>,</span>
<span>  features <span class="op">=</span> <span class="st">"BRAF"</span>,</span>
<span>  sensitivity.measure <span class="op">=</span> <span class="st">"auc_published"</span>,</span>
<span>  molecular.summary.stat <span class="op">=</span> <span class="st">"and"</span>,</span>
<span>  sensitivity.summary.stat <span class="op">=</span> <span class="st">"median"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">sig.mut</span><span class="op">@</span><span class="va">.Data</span><span class="op">[</span>,,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##     estimate       pvalue </span></span>
<span><span class="co">## 8.251857e-01 1.612881e-09</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="univariate-vs-multivariate-analysis">Univariate vs Multivariate Analysis<a class="anchor" aria-label="anchor" href="#univariate-vs-multivariate-analysis"></a>
</h3>
<p>For the final section of this lab, we will explore univariate vs
multivariate linear modeling. Let’s start with a simple univariate
linear model using the same Gene and Drug vector from earlier.</p>
<div class="section level4">
<h4 id="univariate-analysis">Univariate Analysis<a class="anchor" aria-label="anchor" href="#univariate-analysis"></a>
</h4>
<p>Recall <code>Gene</code> is the RNA expression vector of
<code>ENSG00000000005</code> and <code>Drug</code> is the drug response
vector (AUC) for <code>PD-0325901</code> in the GDSC dataset.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">Drug</span> <span class="op">~</span> <span class="va">Gene</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## lm(formula = Drug ~ Gene)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residuals:</span></span>
<span><span class="co">##      Min       1Q   Median       3Q      Max </span></span>
<span><span class="co">## -0.12867 -0.04835 -0.01884  0.05239  0.14848 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Coefficients:</span></span>
<span><span class="co">##              Estimate Std. Error t value Pr(&gt;|t|)</span></span>
<span><span class="co">## (Intercept) -0.004639   0.139300  -0.033    0.974</span></span>
<span><span class="co">## Gene         0.014028   0.017253   0.813    0.443</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residual standard error: 0.08965 on 7 degrees of freedom</span></span>
<span><span class="co">## Multiple R-squared:  0.0863, Adjusted R-squared:  -0.04423 </span></span>
<span><span class="co">## F-statistic: 0.6612 on 1 and 7 DF,  p-value: 0.4429</span></span></code></pre>
<p>Let’s break down this output:</p>
<p><em>Residuals</em>: the difference between the predicted y (drug
response) values and the actual y values. A larger range suggests the
model may not be a good fit of the data.</p>
<p><em>Coefficients</em>: the parameters of the model. -
<code>Estimate</code> can be considered the effect size. Specifically,
it is the one-unit change in the y variable (drug response) per one-unit
change in the x variable (gene expression). The estimate from our
drug~gene association is <code>0.014028</code>. -
<code>Pr(&gt;|t|)</code> is the corresponding p-value for each estimate.
The p-value from our drug~gene association test is
<code>0.443</code>.</p>
<p>Note: If any of the coefficients had met the p-value &lt; 0.05
significance threshold, you would have seen <code>*</code> beside the
estimate (unfortunately, this association does not meet this
threshold).</p>
<p>Let’s see if we can make a multi-variate model with better
performance.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get another gene vector</span></span>
<span><span class="va">commonGenes</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "ENSG00000000005"</span></span></code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Gene2</span> <span class="op">&lt;-</span> <span class="va">GDSCexpression</span><span class="op">[</span><span class="va">commonGenes</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co"># multi-variate model</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">Drug</span> <span class="op">~</span> <span class="va">Gene</span> <span class="op">+</span> <span class="va">Gene2</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Call:</span></span>
<span><span class="co">## lm(formula = Drug ~ Gene + Gene2)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residuals:</span></span>
<span><span class="co">##      Min       1Q   Median       3Q      Max </span></span>
<span><span class="co">## -0.13292 -0.04715 -0.02147  0.04840  0.14047 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Coefficients:</span></span>
<span><span class="co">##             Estimate Std. Error t value Pr(&gt;|t|)</span></span>
<span><span class="co">## (Intercept)  0.15618    0.84445   0.185    0.859</span></span>
<span><span class="co">## Gene         0.01251    0.02017   0.620    0.558</span></span>
<span><span class="co">## Gene2       -0.03441    0.17780  -0.194    0.853</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Residual standard error: 0.09653 on 6 degrees of freedom</span></span>
<span><span class="co">## Multiple R-squared:  0.09197,    Adjusted R-squared:  -0.2107 </span></span>
<span><span class="co">## F-statistic: 0.3038 on 2 and 6 DF,  p-value: 0.7487</span></span></code></pre>
<p>The second gene feature we are adding to our model is
<code>ENSG00000000005</code>.</p>
<p>Take a look at the output, comparing it to the previous model we ran.
Do you think that adding an additional gene has improved model
performance?</p>
</div>
</div>
<div class="section level3">
<h3 id="individual-practice">Individual Practice<a class="anchor" aria-label="anchor" href="#individual-practice"></a>
</h3>
<p>For the remainder of the lab, please explore the associations between
other features and drug response. Try using the CCLE pset too!</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jermiah Joseph, Julia Nguyen, Nikta Feizi, Canadian Bioinformatics Workshop,Bioinformatics.ca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
