<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance • CBWWorkshop2024</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CBWWorkshop2024</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9025</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Introduction</h6></li>
    <li><a class="dropdown-item" href="../articles/PGx_2024.html">Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance</a></li>
    <li><a class="dropdown-item" href="../articles/Workshop_Outline.html">Workshop Outline and Objectives</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Module Labs</h6></li>
    <li><a class="dropdown-item" href="../articles/Module1.html">Module 1: Introduction to pharmacogenomics</a></li>
    <li><a class="dropdown-item" href="../articles/Module2.html">Module 2: Hands-on with pharmacogenomics data</a></li>
    <li><a class="dropdown-item" href="../articles/Module3.html">Module 3: Pharmacogenomics for biomarker discovery – Basic analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Module4.html">Module 4: Pharmacogenomics for biomarker discovery - Advanced analysis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/bhklab/CBWWorkshop2024"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance</h1>
                        <h4 data-toc-skip class="author">Jermiah J.
Joseph</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/bhklab/CBWWorkshop2024/blob/main/vignettes/PGx_2024.Rmd" class="external-link"><code>vignettes/PGx_2024.Rmd</code></a></small>
      <div class="d-none name"><code>PGx_2024.Rmd</code></div>
    </div>

    
    
<!-- Remove {-} from header to show numbering -->
<div class="section level2">
<h2 class="unnumbered" id="workshop-overview">Workshop Overview<a class="anchor" aria-label="anchor" href="#workshop-overview"></a>
</h2>
<div class="section level3">
<h3 class="unnumbered" id="instructors-names-and-contact-information">Instructor(s) name(s) and contact
information<a class="anchor" aria-label="anchor" href="#instructors-names-and-contact-information"></a>
</h3>
<ul>
<li>Jermiah J. Joseph <a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 class="unnumbered" id="workshop-description">Workshop Description<a class="anchor" aria-label="anchor" href="#workshop-description"></a>
</h3>
<!-- Alias package links -->
<p>This workshop will introduce users to the <em><a href="https://bioconductor.org/packages/3.19/CoreGx" class="external-link">CoreGx</a></em> and
<em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
R packages, which are useful tools for pharmacogenomic modelling to
discover biomarkers of treatment response in cancer model systems.
<em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
specifically focuses on drug sensitivity experiments in cancer cell
lines, which will be the major focus of this workshop. Additional
infrastructure from our lab includes <em><a href="https://bioconductor.org/packages/3.19/ToxicoGx" class="external-link">ToxicoGx</a></em>
for toxicogenomics in healthy human cell-lines, <em><a href="https://bioconductor.org/packages/3.19/RadioGx" class="external-link">RadioGx</a></em>
for radiogenomics in cancer cell-lines and <em><a href="https://bioconductor.org/packages/3.19/Xeva" class="external-link">Xeva</a></em> for
pharmacogenomics in patient derived xenograph (PDX) murine models.</p>
<p>Participants will learn the fundamentals of using <em><a href="https://bioconductor.org/packages/3.19/CoreGx" class="external-link">CoreGx</a></em> and
<em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
to create a <code>PharmacoSet</code>—an integrative container for the
storage, analysis and visualization of pharmacogenomic experiments.
Particular focus will be placed on newly developed support for storing,
analyzing and visualizing drug combination sensitivity experiments and
correlating results therefrom with multi-omic molecular profiles to
discover biomarkers of drug sensitivity, resistance, synergy, or
antagonism.</p>
<div class="section level4">
<h4 class="unnumbered" id="pre-requisites">Pre-requisites<a class="anchor" aria-label="anchor" href="#pre-requisites"></a>
</h4>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Knowledge of or interest in pharmacogenomics</li>
<li>Familiarity with core Bioconductor classes such as the and <em><a href="https://bioconductor.org/packages/3.19/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
and <em><a href="https://bioconductor.org/packages/3.19/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
</li>
<li>Curiosity about or experience with <em><a href="https://CRAN.R-project.org/package=data.table" class="external-link">data.table</a></em>
an asset!</li>
</ul>
<p>Useful publications:</p>
<ul>
<li>Smirnov, P., Safikhani, Z., El-Hachem, N., Wang, D., She, A., Olsen,
C., Freeman, M., Selby, H., Gendoo, D. M. A., Grossmann, P., Beck, A.
H., Aerts, H. J. W. L., Lupien, M., Goldenberg, A., &amp; Haibe-Kains,
B. (2016). PharmacoGx: An R package for analysis of large
pharmacogenomic datasets. Bioinformatics (Oxford, England), 32(8),
1244–1246. <a href="https://doi.org/10.1093/bioinformatics/btv723" class="external-link uri">https://doi.org/10.1093/bioinformatics/btv723</a>
</li>
<li>Tonekaboni, M., Ali, S., Soltan Ghoraie, L., Manem, V. S. K. &amp;
Haibe-Kains, B. Predictive approaches for drug combination discovery in
cancer. Brief Bioinform 19, 263–276 (2018).</li>
</ul>
</div>
<div class="section level4">
<h4 class="unnumbered" id="workshop-participation">Workshop Participation<a class="anchor" aria-label="anchor" href="#workshop-participation"></a>
</h4>
<!-- The workshop will be presented as a set of analysis steps to be replicated by the learners, with instructors
available to explain the why and how of applying these functions to the given datasets. Learners will write analysis
scripts as well as use interactive commands to explore the generated data structures and results. Learners will then
brainstorm potential applications of the analysis results in their field as well as comment on use case examples
presented by the instructors on research in our lab. -->
<!-- 
Participants expected to have the following required packages installed on their
machines to be able to run the commands along with the instructors: -->
<!-- ### _R_ / _Bioconductor_ packages used {-}

Bioconductor:

* *[CoreGx](https://bioconductor.org/packages/3.19/CoreGx)*
* *[PharmacoGx](https://bioconductor.org/packages/3.19/PharmacoGx)*

CRAN:

* *[data.table](https://CRAN.R-project.org/package=data.table)*
* *[ggplot2](https://CRAN.R-project.org/package=ggplot2)* -->
<!-- ## Time outline {-}

For a 1.5-hr workshop:

| Activity                                                   | Time |
|------------------------------------------------------------|------|
| Introduction to CoreGx and PharmacoGx                      |  5m  |
| Overview of Data Structures                                |  15m |
| Loading a `PharmacoSet` using *[PharmacoGx](https://bioconductor.org/packages/3.19/PharmacoGx)* or [orcestra.ca](https://orcestra.ca) |  10m |
| Accessing slots of a `PharmacoSet` using *[PharmacoGx](https://bioconductor.org/packages/3.19/PharmacoGx)*           |  10m | -->
<!-- | How the `TRE` Supports Drug Combinations Data Analysis     |  10m |
| Using a `DataMapper` to build a Drug Combo PharmacoSet     |  10m |
| Dose Response Models and Drug Sensitivity measures         |  10m |
| Drug Combination Synergy Models                            |  15m |
| Biomarker Discovery                                        |  10m |
| Further Resources                                          |  5m  |
 -->
<!-- 
## Workshop goals and objectives {-}

### Learning goals {-}

* Describe pharmacogenomic mono and combination datasets and their usefulness in cancer research
* Understand how experimental designs and research questions map onto data structures
* Learn how to extract information from these datasets
* Learn how to visualize experimental results from these datasets
* Learn how to model dose-response for both monotherapy and combination small compound datasets
* Learn measures to quantify response and synergy in cancer cell-line sensitivity screens

### Learning objectives {-}

* Describe use cases for *[CoreGx](https://bioconductor.org/packages/3.19/CoreGx)* and *[PharmacoGx](https://bioconductor.org/packages/3.19/PharmacoGx)* in pharmacogenomics and beyond
* Understand the structure of the `CoreSet` and `PharmacoSet` classes to facilitate their use in subsequent analyses
* Learn how the `TreatmentResponseExperiment` provides a highly flexible container for storing high-throughput drug combination screening data
* Use a `TREDataMapper` to create a `TreatmentResponseExperiment` and incorporate it into a `PharmacoSet`
* Learn to apply arbitrary R functions over a `TreatmentResponseExperiment` using the `aggregate` and `endoaggregate` methods
* Access the molecular features, dose-response and metadata contained within the data structures defined in our packages
* Fit Hill Slope models to dose-response experiments using small compound treatments in cell-lines
* Calculate the AAC, AUC, and IC50 metrics for response quantification in cell lines
* Compute and store various metrics of drug synergy and antagonism using *[PharmacoGx](https://bioconductor.org/packages/3.19/PharmacoGx)* and the `TreatmentResponseExperiment`
* Learn how to use `PharmacoGx` to discover univariate biomarkers of drug response, resistance, synergy, or antagnoism

---- -->
</div>
</div>
</div>
<div class="section level2">
<h2 id="introduction-to-coregx-and-pharmacogx">Introduction to CoreGx and PharmacoGx<a class="anchor" aria-label="anchor" href="#introduction-to-coregx-and-pharmacogx"></a>
</h2>
<p>This tutorial, titled <em>Pharmacogenomic Analysis of Drug
Combination Experiments to Identify Biomarkers of Response or
Resistance</em>, focuses on using the <em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
R package to correlate treatment response, measured as the viability of
cancer cell-lines after <em>in vitro</em> drug treatment, with their
respective multi-omic profiles. <em><a href="https://bioconductor.org/packages/3.19/CoreGx" class="external-link">CoreGx</a></em>
provides the core infrastructure for storing, analyzing, and visualizing
generic treatment response experiments. It provides methods and classes
which can be inherited from in downstream packages, such as <em><a href="https://bioconductor.org/packages/3.19/ToxicoGx" class="external-link">ToxicoGx</a></em>
and <em><a href="https://bioconductor.org/packages/3.19/RadioGx" class="external-link">RadioGx</a></em>.
We hope that the <code>CoreSet</code> object is generalized enough that
it can be reused by other developers for their specific treatment (or
stimuli) response use case.</p>
<div class="section level3">
<h3 id="coregx">CoreGx<a class="anchor" aria-label="anchor" href="#coregx"></a>
</h3>
<div class="section level4">
<h4 id="package-nomenclature">Package Nomenclature<a class="anchor" aria-label="anchor" href="#package-nomenclature"></a>
</h4>
<p>To facilitate modularization of the GxSuite of R packages, we have
shifted the nomenclature within a <code>CoreSet</code>—and therefore in
inheriting packages —to be more general.</p>
<p>To this end, have made the following changes:</p>
<ul>
<li>Previous reference to cell (cell-line) have become sample, allowing
the <code>CoreSet</code> to be used for other model systems</li>
<li>Drug (radiation in <code>RadioGx</code>) have become treatment,
allowing the <code>CoreSet</code> to be treatment type (or stimuli)
agnostic</li>
<li>Sensitivity will become response (sensitivity slot becomes
treatmentResponse)</li>
</ul>
<p>As a result of these changes, the names of some common accessors have
been updated. The old accessors still remain functional to ensure
backwards compatibility for at least two Bioconductor releases. A
deprecation warning will be added to old accessors informing users of
the corresponding new function, as per Bioconductor best practices.</p>
</div>
</div>
<div class="section level3">
<h3 id="pharmacogx">PharmacoGx<a class="anchor" aria-label="anchor" href="#pharmacogx"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.19/PharmacoGx" class="external-link">PharmacoGx</a></em>
stores drug screening data together with molecular profiling of
cell-lines in an object called a <code>PharmacoSet</code>, or
<code>PSet</code> for short.</p>
<p>Previously, the <code>PharmacoSet</code> class was entirely defined
by the <code>PharmacoGx</code> package. However, after building the
<code>RadioGx</code> and <code>ToxicoGx</code> packages, we realized
that the core data structures could be abstracted out and shared.</p>
<p>As such, <code>PharmacoSet</code>s now inherit from the
<code>CoreSet</code> class defined in our package <code>CoreGx</code>,
which is used to share common datas structure and method across our
suite of package. The primary use case for <code>PharmacoGx</code> is to
:</p>
<ol style="list-style-type: decimal">
<li>Provide a standardized and highly curated container for
high-throughput screens in cancer-cell lines</li>
<li>Enable discovery of biomarkers of treatment response or
resistance</li>
<li>Allow for comparison and validation across large published
pharmacogneomic datasets</li>
</ol>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="overview-of-data-structures">Overview of Data Structures<a class="anchor" aria-label="anchor" href="#overview-of-data-structures"></a>
</h2>
<p>The GxSuite of packages make use of various Bioconductor classes for
storing molecular profile data.</p>
<div class="section level3">
<h3 id="coreset-and-pharmacoset">
<code>CoreSet</code> and <code>PharmacoSet</code><a class="anchor" aria-label="anchor" href="#coreset-and-pharmacoset"></a>
</h3>
<div class="float">
<img src="CoreSet-class.drawio.svg" alt="CoreSet / PharmacoSet class diagram. Objects comprising a CoreSet are enclosed in boxes. The first box indicates the type and name of each object. The second box indicates the structure of an object or class. The third box shows accessor methods from PharmacoGx for that specific object. ‘=&gt;’ represents return and specifies what is returned from that item or method."><div class="figcaption">
<strong>CoreSet / PharmacoSet class
diagram</strong>. Objects comprising a <code>CoreSet</code> are enclosed
in boxes. The first box indicates the type and name of each object. The
second box indicates the structure of an object or class. The third box
shows accessor methods from <code>PharmacoGx</code> for that specific
object. ‘=&gt;’ represents return and specifies what is returned from
that item or method.</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="example-experiment">Example Experiment<a class="anchor" aria-label="anchor" href="#example-experiment"></a>
</h2>
<div class="section level3">
<h3 id="cell-lines-samples">Cell Lines (<code>samples</code>)<a class="anchor" aria-label="anchor" href="#cell-lines-samples"></a>
</h3>
<p>The following cell lines are used in the study:</p>
<ul>
<li>
<strong>Cell Line A (CLC-123)</strong>: A breast cancer cell
line.</li>
<li>
<strong>Cell Line B (CLC-456)</strong>: A lung cancer cell
line.</li>
<li>
<strong>Cell Line C (CLC-789)</strong>: A colon cancer cell
line.</li>
<li>
<strong>Cell Line D (CLC-101)</strong>: A melanoma cell line.</li>
<li>
<strong>Cell Line E (CLC-202)</strong>: A pancreatic cancer cell
line.</li>
</ul>
</div>
<div class="section level3">
<h3 id="drugs-treatments">Drugs (<code>treatments</code>)<a class="anchor" aria-label="anchor" href="#drugs-treatments"></a>
</h3>
<p>The following drugs were tested on the cell lines:</p>
<ul>
<li>
<strong>Drug X (Toxo-1)</strong>: A novel chemotherapy drug that
targets fast-dividing cancer cells.</li>
<li>
<strong>Drug Y (Inhi-2)</strong>: A targeted therapy designed to
block a protein that is commonly overactive in certain cancers.</li>
<li>
<strong>Drug Z (Syner-3)</strong>: A combination therapy that
enhances the effectiveness of Drug X when used together.</li>
</ul>
</div>
<div class="section level3">
<h3 id="profiles-of-the-samples-molecularprofiles">Profiles of the samples (<code>molecularProfiles</code>)<a class="anchor" aria-label="anchor" href="#profiles-of-the-samples-molecularprofiles"></a>
</h3>
<p>We managed to obtain the following molecular profiles for each of the
cell lines:</p>
<ol style="list-style-type: decimal">
<li>RNA Expression (rnaseq)</li>
<li>DNA Methylation (methylation)</li>
<li>Mutations (mutation)</li>
<li>Copy Number Variation (cnv)</li>
</ol>
</div>
<div class="section level3">
<h3 id="treatment-responses-treatmentresponse">Treatment Responses (<code>treatmentResponse</code>)<a class="anchor" aria-label="anchor" href="#treatment-responses-treatmentresponse"></a>
</h3>
<p>For each sample, the treatments were tested with the following
doses:</p>
<ul>
<li>1 µM</li>
<li>10 µM</li>
<li>100 µM</li>
<li>1000 µM</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jermiah Joseph, Julia Nguyen, Nikta Feizi, Canadian Bioinformatics Workshop,Bioinformatics.ca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
