<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Workshop Outline and Objectives • CBWWorkshop2024</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Workshop Outline and Objectives">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CBWWorkshop2024</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9019</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Introduction</h6></li>
    <li><a class="dropdown-item" href="../articles/PGx_2024.html">Pharmacogenomic Analysis of Drug Combination Experiments to Identify Biomarkers of Response or Resistance</a></li>
    <li><a class="dropdown-item" href="../articles/Workshop_Outline.html">Workshop Outline and Objectives</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Module Labs</h6></li>
    <li><a class="dropdown-item" href="../articles/Module1.html">Module 1: Introduction to pharmacogenomics</a></li>
    <li><a class="dropdown-item" href="../articles/Module2.html">Module 2: Hands-on with pharmacogenomics data</a></li>
    <li><a class="dropdown-item" href="../articles/Module3.html">Module 3: Intro to pharmacogenomics in the context of cancer</a></li>
    <li><a class="dropdown-item" href="../articles/Module4.html">Module 4: Pharmacogenomics for biomarker discovery - Advanced analysis</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/bhklab/CBWWorkshop2024"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Workshop Outline and Objectives</h1>
                        <h4 data-toc-skip class="author">Jermiah J.
Joseph</h4>
            <address class="author_afil">
      Princess Margaret Cancer
Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:jermiah.joseph@uhn.ca" class="email">jermiah.joseph@uhn.ca</a>
      </address>
                  
            <h4 data-toc-skip class="date">16 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/bhklab/CBWWorkshop2024/blob/main/vignettes/Workshop_Outline.Rmd" class="external-link"><code>vignettes/Workshop_Outline.Rmd</code></a></small>
      <div class="d-none name"><code>Workshop_Outline.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 class="unnumbered" id="course-description">Course Description<a class="anchor" aria-label="anchor" href="#course-description"></a>
</h2>
<p>This course explores the evolving field of personalized cancer
treatment, focusing on tailoring treatment strategies to patients based
on their unique molecular profiles. Participants will gain a
comprehensive understanding of the field of pharmacogenomics and a
guided workflow covering basic to advanced biomarker analysis. Our
primary emphasis is on the analysis of cancer biomarkers; however, the
principles and methodologies covered in this context are transferable
and applicable to other disease areas.</p>
<p>The course is designed to provide sufficient context to beginners,
building up from basics to tools and methods of data analysis. The
hands-on practical tutorials for each session will provide the
participants with practical insights into the data as well as tools to
perform biomarker analysis on their own. By the end of this introductory
course, participants will be equipped with the knowledge and skills to
understand data requirements and perform biomarker analysis.</p>
</div>
<div class="section level2">
<h2 id="course-objectives">Course Objectives<a class="anchor" aria-label="anchor" href="#course-objectives"></a>
</h2>
<p>Participants will gain practical experience and skills to be able
to:</p>
<p>Extract features for biomarker discovery from -omics data (RNA-Seq,
methylation, mutations, proteomics, chromatin profiling, etc) Navigate
pharmacogenomics datasets and data preparation for pharmacogenomic
analysis Identify univariable biomarkers from RNA-Seq data Perform
meta-analysis to identify robust biomarkers Visualize and summarize the
output of pharmacogenomic biomarker analysis</p>
</div>
<div class="section level2">
<h2 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h2>
<p>Participants should be comfortable writing and reading basic R.</p>
<p>You will require your own laptop computer. Minimum requirements:
1024×768 screen resolution, 1.5GHz CPU, 4GB RAM, 20GB free disk space,
recent versions of Windows, Mac OS X or Linux (Most computers purchased
in the past 3-4 years likely meet these requirements).</p>
<p>This workshop requires participants to complete pre-workshop tasks
and readings which includes installing and setting up an R
environment.</p>
</div>
<div class="section level2">
<h2 id="course-outline">Course Outline<a class="anchor" aria-label="anchor" href="#course-outline"></a>
</h2>
<div class="section level3">
<h3 id="module-1-introduction-to-pharmacogenomics-and-biomarker-discovery">Module 1: Introduction to Pharmacogenomics and Biomarker
Discovery<a class="anchor" aria-label="anchor" href="#module-1-introduction-to-pharmacogenomics-and-biomarker-discovery"></a>
</h3>
<div class="section level4">
<h4 id="lecture">Lecture:<a class="anchor" aria-label="anchor" href="#lecture"></a>
</h4>
<ul>
<li>A brief introduction to precision oncology and biomarkers
<ul>
<li>Overview to personalized cancer treatment and the role of biomarker
discovery</li>
<li>What is “pharmacogenomics”; is it different from
pharmacogenetics?</li>
<li>Challenges in biomarker discovery</li>
</ul>
</li>
<li>Brief introduction to biomarkers in pharmacogenomics
<ul>
<li>Conventional markers: oncogenic mutations, gene expression,
proteomics</li>
<li>Exciting new fields: mutational signatures, synthetic lethality,
chromatin accessibility</li>
<li>Other markers like radiomic features</li>
</ul>
</li>
<li>Overview of methods for patient profiling
<ul>
<li>Sequencing technologies: RNA-Seq, Whole Genome Sequencing (WGS),
Whole Exome Sequencing (WES), etc.</li>
<li>Solid tumor vs liquid biopsy</li>
</ul>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="practical-getting-to-know-multi-omics-data">Practical – Getting to know multi-omics data<a class="anchor" aria-label="anchor" href="#practical-getting-to-know-multi-omics-data"></a>
</h4>
<p>TAs: Julia Nguyen, Jermiah Joseph</p>
<ul>
<li>Loading in expression matrices for various -omics data (RNA-Seq,
methylation, mutations, proteomics, chromatin profiling)</li>
<li>Exploring feature extraction techniques to define biomarkers from
(epi)genomic data</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="module-2-data-preparation-for-pharmacogenomic-analysis">Module 2: Data Preparation for Pharmacogenomic Analysis<a class="anchor" aria-label="anchor" href="#module-2-data-preparation-for-pharmacogenomic-analysis"></a>
</h3>
<div class="section level4">
<h4 id="lecture-1">Lecture:<a class="anchor" aria-label="anchor" href="#lecture-1"></a>
</h4>
<ul>
<li>Common preclinical pharmacogenomic models and importance of clinical
validation
<ul>
<li>Cell lines, patient-derived xenografts (PDX), organoids, etc.</li>
<li>Importance of clinical validation in biomarker discovery</li>
</ul>
</li>
<li>What is in a typical pharmacogenomic dataset?: dataset overview
using PharmacoGx</li>
<li>Quality control: batch effects, outliers, missing values, and common
sources of variation</li>
</ul>
</div>
<div class="section level4">
<h4 id="practical-hands-on-with-pharmacogenomics-data">Practical: Hands-on with pharmacogenomics data<a class="anchor" aria-label="anchor" href="#practical-hands-on-with-pharmacogenomics-data"></a>
</h4>
<ul>
<li>Loading a PharmacoSet in R using PharmacoGx or ORCESTRA,
understanding the data structure, accessing features, and preparing
metadata</li>
<li>Plot batch effects using PCA and run correction method (SVA)</li>
<li>Filter out outliers and missing values</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="module-3-pharmacogenomics-for-biomarker-discovery-basic-analysis">Module 3: Pharmacogenomics for biomarker discovery – Basic
analysis<a class="anchor" aria-label="anchor" href="#module-3-pharmacogenomics-for-biomarker-discovery-basic-analysis"></a>
</h3>
<div class="section level4">
<h4 id="lecture-2">Lecture:<a class="anchor" aria-label="anchor" href="#lecture-2"></a>
</h4>
<ul>
<li>Data analysis walk-through
<ul>
<li>Statistical methods for drug response associations</li>
<li>Computing drug response associations using Concordance Index and
Pearson’s correlation</li>
<li>Introduction to univariable vs. multivariable analysis for
biomarkers</li>
</ul>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="practical-statistical-analysis">Practical: Statistical Analysis<a class="anchor" aria-label="anchor" href="#practical-statistical-analysis"></a>
</h4>
<ul>
<li>Basic functionalities of PharmacoGx</li>
<li>Case study – Univariable analysis of drug response using
PharmacoGx</li>
<li>Plotting associations and interpretation of results in R</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="module-4-pharmacogenomics-for-biomarker-discovery-advanced-analysis">Module 4: Pharmacogenomics for biomarker discovery – Advanced
analysis<a class="anchor" aria-label="anchor" href="#module-4-pharmacogenomics-for-biomarker-discovery-advanced-analysis"></a>
</h3>
<div class="section level4">
<h4 id="lecture-3">Lecture:<a class="anchor" aria-label="anchor" href="#lecture-3"></a>
</h4>
<ul>
<li>Meta-analysis
<ul>
<li>Introduction to “meta-analysis” in the context of inter-study
investigations</li>
<li>Visualization and interpretation using forest plots</li>
<li>How to interpret meta-analysis statistics</li>
</ul>
</li>
<li>Sub-group analysis
<ul>
<li>Identifying tissue-specific biomarkers with an example of cancer
subtypes</li>
<li>Other factors to consider (tumor grade, biological sex, etc)</li>
</ul>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="practical-statistical-analysis-part-2">Practical: Statistical Analysis – Part 2<a class="anchor" aria-label="anchor" href="#practical-statistical-analysis-part-2"></a>
</h4>
<ul>
<li>Run meta-analysis on ~4 small PharmacoSets for a biomarker of
interest for pan-cancer and cancer-specific data</li>
<li>Generate and interpret forest plots with statistics</li>
</ul>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jermiah Joseph, Julia Nguyen, Nikta Feizi, Canadian Bioinformatics Workshop,Bioinformatics.ca.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
